# Image-related record format

_<mark style="background-color:purple;">\*The interpretation of the Advanced part is only available in EN.</mark>_

## Image processing record (IPR)

Image processing record (IPR) file is designed to record the whole-life information of a microscope image from photo-taking to processing. Each staining image (ssDNA/DAPI, IF, H\&E) includes six basic groups, "ImageInfo", "QCInfo", "Stitch", "TissueSeg", "CellSeg", and "Register", which are used to store microscopy photo-taking information, image quality control information, image stitching records, tissue segmentation records, cell segmentation records (optional), and registration records.&#x20;

<table data-full-width="false"><thead><tr><th>Item</th><th>Description</th></tr></thead><tbody><tr><td><strong>Attribute</strong></td><td><strong>Description</strong></td></tr><tr><td>IPRVersion</td><td>IPR file format version.</td></tr><tr><td><strong>Dateset</strong></td><td><strong>Description</strong></td></tr><tr><td>Preview</td><td>A 2D matrix merges stitched image, tissue segmentation boundary and cell segmentation boundary.</td></tr><tr><td><em><strong>/&#x3C;stainType>/ImageInfo: Group records basic image information.</strong></em></td><td></td></tr><tr><td><strong>Attributes</strong></td><td><strong>Description</strong></td></tr><tr><td>AppFileVer</td><td>Microscope software version.</td></tr><tr><td>BackgroundBalance</td><td>Background balance.</td></tr><tr><td>BitDepth</td><td>The bit-depth of a camera sensor describes its ability to transform the analog signal coming from the pixel array into a digital signal.</td></tr><tr><td>Brightness</td><td>Relative intensity affecting a person or sensor.</td></tr><tr><td>ChannelCount</td><td>Number of RGB channels.</td></tr><tr><td>ColorEnhancement</td><td>Whether enhanced image color display or not.</td></tr><tr><td>Contrast</td><td>The difference in color and intensity of the depicted object from its background.</td></tr><tr><td>DeviceSN</td><td>Microscope device serial number.</td></tr><tr><td>DistortionCorrection</td><td>Whether fixed distortion or not.</td></tr><tr><td>ExposureTime</td><td>Exposure time in ms.</td></tr><tr><td>FOVHeight</td><td>Height of an individual FOV in pixel.</td></tr><tr><td>FOVWidth</td><td>Width of an individual FOV in pixel.</td></tr><tr><td>Gain</td><td>Amplification applied to the signal by the image sensor.</td></tr><tr><td>Gamma</td><td>The coefficient links between the human eye and the digital camera.</td></tr><tr><td>GammaShift</td><td>Whether adapt the digital image taken with the help of a linearly recording camera to the nonlinear perception of the human eye or not.</td></tr><tr><td>Illuminance</td><td>Intensity of light.</td></tr><tr><td>Manufacturer</td><td>Microscope manufacturer.</td></tr><tr><td>Model</td><td>Microscope model.</td></tr><tr><td>Overlap</td><td>Overlapping pixels between single tiles.</td></tr><tr><td>PixelSizeX</td><td>Size of pixel in x direction.</td></tr><tr><td>PixelSizeY</td><td>Size of pixel in y direction.</td></tr><tr><td>QCResultFile</td><td>Prefix of ImageQC/ImageStudio result file, the unique identifier of the image.</td></tr><tr><td>ScanChannel</td><td>Fluorescence channel.</td></tr><tr><td>ScanCols</td><td>Number of columns scanned.</td></tr><tr><td>ScanObjective</td><td>Magnification power of the scan objective lens.</td></tr><tr><td>ScanRows</td><td>Number of rows scanned.</td></tr><tr><td>ScanTime</td><td>Scan date and time.</td></tr><tr><td>Sharpness</td><td>Degree of clarity of the edge(s) of the image.</td></tr><tr><td>RegisterVersion</td><td>Image auto-processing &#x26; matrix registration software version.</td></tr><tr><td>StitchedImage</td><td>Whether the corresponding image is a panorama image (true) or a set of tiled images (false).</td></tr><tr><td>STOmicsChipSN</td><td>Stereo-seq Chip serial number.</td></tr><tr><td>WhiteBalance</td><td>An adjustment in electronic and film imaging that corrects the color balance of the lighting.</td></tr><tr><td><strong>DatasetDataType: 1D array</strong></td><td><strong>Description</strong></td></tr><tr><td>RGBScale</td><td>RGB color.</td></tr><tr><td><em><strong>/&#x3C;stainType>/QCInfo: Group records the QC information of the image.</strong></em></td><td></td></tr><tr><td><strong>Attributes</strong></td><td><strong>Description</strong></td></tr><tr><td>ClarityScore</td><td>Reference score for evaluating the clearness of cell boundaries.</td></tr><tr><td>Experimenter</td><td>Email of the experimenter who did QC for the image.</td></tr><tr><td>GoodFOVCount</td><td>Number of FOVs that have identified more than 3 track cross points.</td></tr><tr><td>ImageQCVersion</td><td>Image  QC version.</td></tr><tr><td>QCPassFlag</td><td>Whether the corresponding image passed QC.</td></tr><tr><td>RemarkInfo</td><td>Any remarks, notes, comments on the image.</td></tr><tr><td>StainType</td><td>Stain type of microscopy image.</td></tr><tr><td>TotalFOVCount</td><td>Total number of FOVs.</td></tr><tr><td>TrackLineScore<br></td><td>Reference score for evaluating whether the detected track lines can be used as references for image stitching and registering with gene expression matrix. (This score only evaluate whether the program detected track lines on the image, it does not infer the clarity of the lines or the images).</td></tr><tr><td>TrackLineChannel</td><td>Shooting channel of track lines.</td></tr><tr><td>TrackCrossQCPassFlag</td><td>Whether the QC of the track-cross point is passed.</td></tr><tr><td>ScopeStitchQCScore</td><td>Score to assess the stability of microscope overlap (DAPI &#x26; mIF only).</td></tr><tr><td>ScopeStitchQCPassFlag</td><td>Whether the QC of microscope stitch is passed (DAPI &#x26; mIF only).</td></tr><tr><td>TemplateValidArea</td><td>Proportion of encircled area by the detected points that match the global track line template (within the error range of 5 pixels), to the entire image area.</td></tr><tr><td>TemplateRecall</td><td>Proportion of detected points, which match the global track line template (within the error range of 5 pixels), to the points derived from the chip track line rules.</td></tr><tr><td>ClarityCutSize</td><td>Size of cut images (based on FOV) for clarity evaluation.</td></tr><tr><td>ClarityCounts</td><td>Counts of cut images in different categories.</td></tr><tr><td>ClarityOverlap</td><td>Overlap of cut images for clarity evaluation.</td></tr><tr><td><strong>DatasetDataType: 2D array</strong></td><td><strong>Description</strong><br></td></tr><tr><td>ScopeStitchQCMatrix</td><td>Matrix of overlap deviation of each microscope FOV (DAPI &#x26; mIF only).</td></tr><tr><td>CrossPoints/row_col*n<br></td><td>Group of datasets for each FOV that records the track cross point coordinates. (Row and col stand for the FOV row and column index number, and n stands for number of FOVs). Each dataset is a 2D array record, (x, y) coordinates of track cross points in each FOV.</td></tr><tr><td><em><strong>/&#x3C;stainType>/Calibration: Group records the calibration information (&#x3C;protein>_IF only).</strong></em></td><td></td></tr><tr><td><strong>Attributes</strong></td><td><strong>Description</strong></td></tr><tr><td>CalibrationQCPassFlag</td><td>Whether the calibration QC of IF image is passed.</td></tr><tr><td><strong>/&#x3C;stainType>/Calibration/Scope:</strong></td><td></td></tr><tr><td><strong>Attributes</strong></td><td><strong>Description</strong></td></tr><tr><td>Confidence</td><td>Calibration confidence of microscope-stitched tiled image vs. IF.</td></tr><tr><td>OffsetX</td><td>Horizontal offset.</td></tr><tr><td>OffsetY</td><td>vertical offset.</td></tr><tr><td><strong>[optional]/&#x3C;stainType>/Calibration/Stereo:</strong></td><td></td></tr><tr><td><strong>Attributes</strong></td><td><strong>Description</strong></td></tr><tr><td>Confidence</td><td>Calibration confidence of Stereo-stitched image vs. IF.</td></tr><tr><td>OffsetX</td><td>Horizontal offset.</td></tr><tr><td>OffsetY</td><td>Vertical offset.</td></tr><tr><td><em><strong>/&#x3C;stainType>/Stitch: Group records the stitching information.</strong></em></td><td></td></tr><tr><td><strong>Attributes</strong></td><td><strong>Description</strong></td></tr><tr><td>StitchingScore</td><td>Reference score for stitching.</td></tr><tr><td>TemplateSource</td><td>The reference FOV for deriving the template used for rotating and scaling the microscopic images.</td></tr><tr><td>WhichStitch</td><td>Stitching method for image, including microscope, template and ripple stitching. The default for stitched image is microscope.</td></tr><tr><td><strong>DatasetDataType: 2D array</strong></td><td><strong>Description</strong></td></tr><tr><td>TemplatePoint<br></td><td>Center coordinates for deriving template lines.</td></tr><tr><td>TransformTemplate</td><td>Coordinates of template points registered with expression matrix.</td></tr><tr><td><em><strong>/&#x3C;stainType>/Stitch/StereoStitch: Group records the image stitching information processed by Stereo program.</strong></em></td><td></td></tr><tr><td><strong>Attributes</strong></td><td><strong>Description</strong></td></tr><tr><td>StitchedGlobalHeight</td><td>Height of stitched tiled images using Stereo stitching algorithm. Tiled image only.</td></tr><tr><td>StitchedGlobalWidth</td><td>Width of stitched tiled images using Stereo stitching algorithm. Tiled image only.</td></tr><tr><td><strong>DatasetDataType: 2D array</strong></td><td><strong>Description</strong></td></tr><tr><td>StitchedGlobalLoc</td><td>Coordinates for the Stereo stitched tiled image. Tiled image only.</td></tr><tr><td><em><strong>/&#x3C;stainType>/Stitch/ScopeStitch: Group records the image stitching information processed by microscope imaging software.</strong></em></td><td></td></tr><tr><td><strong>Attributes</strong></td><td><strong>Description</strong></td></tr><tr><td>GlobalHeight</td><td>Height of panorama image.</td></tr><tr><td>GlobalWidth</td><td>Width of panorama image.</td></tr><tr><td><strong>DatasetDataType: 2D array</strong></td><td><strong>Description</strong></td></tr><tr><td>GlobalLoc</td><td>Coordinates for the stitched tiled image (either program stitched or microscope stitched).</td></tr><tr><td>ScopeJitterDiff</td><td>Jitter offset of microscope stitching.</td></tr><tr><td>ScopeHorizontalJitter</td><td>Horizontal jitter offset of microscope stitching. Tiled image only.</td></tr><tr><td>ScopeVerticalJitter</td><td>Vertical jitter offset of microscope stitching. Tiled image only.</td></tr><tr><td><em><strong>/&#x3C;stainType>/Stitch/StitchEval: Group records the evaluation result of stitching.</strong></em></td><td></td></tr><tr><td><strong>Attribute</strong></td><td><strong>Description</strong></td></tr><tr><td>MaxDeviation</td><td>Maximum stitching deviation.</td></tr><tr><td><strong>DatasetDataType: 2D array</strong></td><td><strong>Description</strong></td></tr><tr><td>GlobalDeviation</td><td>Global stitching deviation matrix. Tiled image only.</td></tr><tr><td>StitchEvalH</td><td>Stitching deviation matrix for the horizontal axes.</td></tr><tr><td>StitchEvalV</td><td>Stitching deviation matrix for the vertical axes.</td></tr><tr><td><em><strong>/&#x3C;stainType>/Register: Group records the information that aligns images with gene expression matrix.</strong></em></td><td></td></tr><tr><td><strong>Attributes</strong></td><td><strong>Description</strong></td></tr><tr><td>CounterRot90</td><td>Count of counter-clockwise rotation of 90 degrees.</td></tr><tr><td>Flip</td><td>Whether horizontally flipped or not.</td></tr><tr><td>MatrixShape</td><td>Height and width of the gene expression matrix.</td></tr><tr><td>OffsetX</td><td>Offset between microscope image and gene expression matrix in x-axis.</td></tr><tr><td>OffsetY</td><td>Offset between microscope image and gene expression matrix in y-axis.</td></tr><tr><td>RegisterScore</td><td>Reference score for registration.</td></tr><tr><td>Rotation</td><td>Rotation degree between raw image and deviation template.</td></tr><tr><td>ScaleX</td><td>Scale between raw image and deviation template in horizontal direction.</td></tr><tr><td>ScaleY</td><td>Scale between raw image and deviation template in vertical direction.</td></tr><tr><td>XStart</td><td>Gene expression matrix offset x (GEF geneExp/binN/expression attribute minX).</td></tr><tr><td>YStart</td><td>Gene expression matrix offset y (GEF geneExp/binN/expression attribute minY).</td></tr><tr><td>ManualRotation</td><td>Manual rotation degree of the raw image around the center point.</td></tr><tr><td>ManualScaleX</td><td>Manual scale of the raw image in horizontal direction based on center point.</td></tr><tr><td>ManualScaleY</td><td>Manual scale of the raw image in vertical direction based on center point.</td></tr><tr><td><strong>DatasetDataType: 2D array</strong></td><td><strong>Description</strong></td></tr><tr><td>MatrixTemplate</td><td>List of track cross points derived from gene expression matrix.</td></tr><tr><td><em><strong>/&#x3C;stainType>/TissueSeg: Group records the tissue segmentation information.</strong></em></td><td></td></tr><tr><td><strong>Attributes</strong></td><td><strong>Description</strong></td></tr><tr><td>TissueSegScore</td><td>Reference score for tissue segmentation.</td></tr><tr><td>TissueSegShape</td><td>Image shape for tissue segmentation mask image.</td></tr><tr><td>Threshold</td><td>The threshold for tissue segmentation in IF images is divided into maximum and minimum values (IF only).</td></tr><tr><td><strong>DatasetDataType: 2D array</strong></td><td><strong>Description</strong></td></tr><tr><td>TissueMask</td><td>Encoded tissue segmentation mask file (after registration with gene expression matrix).</td></tr><tr><td><em><strong>/&#x3C;stainType>/CellSeg: Group records the cell segmentation information.</strong></em></td><td></td></tr><tr><td><strong>Attributes</strong></td><td><strong>Description</strong></td></tr><tr><td>CellSegShape</td><td>Image shape for cell segmentation mask image.</td></tr><tr><td><strong>DatasetDataType: 2D array</strong></td><td><strong>Description</strong></td></tr><tr><td>CellMask</td><td>Encoded cell segmentation mask file (after registration with gene expression matrix).</td></tr><tr><td>CellSegTrace</td><td>Cell contour attributes, including height, width and area.</td></tr><tr><td><em><strong>/StereoResepSwitch: Group stores the state of each module that whether the module needs to be performed.</strong></em></td><td></td></tr><tr><td><strong>Attributes</strong></td><td><strong>Description</strong></td></tr><tr><td>stitch</td><td>Switch for performing stitching.</td></tr><tr><td>tissueseg</td><td>Switch for performing tissue segmentation.</td></tr><tr><td>cellseg</td><td>Switch for performing cell segmentation.</td></tr><tr><td>register</td><td>Switch for performing registration.</td></tr><tr><td><em><strong>/ManualState: Group stores the state of each module that whether the module has been manually processed.</strong></em></td><td></td></tr><tr><td><strong>Attributes</strong></td><td><strong>Description</strong></td></tr><tr><td>stitch</td><td>Whether manually stitched the tiled images.</td></tr><tr><td>tissueseg</td><td>Whether manually delineated the tissue coverage region.</td></tr><tr><td>cellseg</td><td>Whether manually delineated the cell coverage regions.</td></tr><tr><td>register</td><td>Whether manually aligned microscope image and gene expression matrix.</td></tr><tr><td>calibration</td><td>Whether manually calibrated two images. For example, manually match IF image with DAPI image.</td></tr></tbody></table>

## Recorded image processing (RPI)

The image pyramid model is a multi-resolution hierarchical model that is used to store and display images in different resolutions. For the same field of view, the layer of the image pyramid that is closest to the bottom includes the most detailed information and has the largest scale. `register` pipeline performs the down-sampling step on the registered image, and the result images are layered to construct a pyramid with the suffix ".rpi". For each resolution layer, the intact registered image is split into 256 pixels x 256 pixels tiles. If the size of a layer is smaller than 256 x 256, the image will then remain intact. The outer layer group of RPI file is defined according to its stain type, generally including ssDNA, DAPI, IF (immunofluorescence image especially for protein, group name as `<protein_IF>`), and H\&E. In each group of stain types, multiple image results, including an image subgroup (registered microscopy image), a `TissueMask` subgroup (registered mask boundary for the tissue coverage area), and a `CellMask` subgroup (registered mask boundaries for the cell coverage area, optional), could be saved respectively.

Schematic diagram of the image pyramid:

<figure><img src="../img/assets/image (17).png" alt=""><figcaption><p>Image pyramid displaying</p></figcaption></figure>

| Item                                                                                                                                                          | Description                                        |
| ------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------------------------------------------------- |
| **Attributes**                                                                                                                                                | **Description**                                    |
| imgSize                                                                                                                                                       | Size of the image.                                 |
| sizex                                                                                                                                                         | Max of x.                                          |
| sizey                                                                                                                                                         | Max of y.                                          |
| version                                                                                                                                                       | RPI file version.                                  |
| x\_start                                                                                                                                                      | Offset x of the matrix.                            |
| y\_start                                                                                                                                                      | Offset y of the matrix.                            |
| _**/\<stainType>/\<imageLayer>: Group records the specific layer information, mainly including "Image", "TissueMask", "CellMask" and "CellMask\_adjusted".**_ |                                                    |
| **Attributes**                                                                                                                                                | **Description**                                    |
| MaxGrayLevel                                                                                                                                                  | Maximum gray value.                                |
| MinGrayLevel                                                                                                                                                  | Minimum gray value.                                |
| GrayLevelElbow                                                                                                                                                | Elbow point of gray value.                         |
| TrackLayer                                                                                                                                                    | Whether track layer.                               |
| Color                                                                                                                                                         | Layer color.                                       |
| **Groups**                                                                                                                                                    | **Description**                                    |
| binN                                                                                                                                                          | Detailled downsampling metadata for each bin size. |

