
<!DOCTYPE HTML>
<html lang="zh-hans">
    <head>
        <meta charset="UTF-8">
        <meta content="text/html; charset=utf-8" http-equiv="Content-Type">
        <title>HTML report &#xB7; SAW User Manual</title>
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="description" content>
        <meta name="generator" content="GitBook 3.2.3">
        
        
        
    
    <link rel="stylesheet" href="../../gitbook/style.css">

    
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-expandable-chapters-small/expandable-chapters-small.css">
                
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-anchor-navigation-ex/style/plugin.css">
                
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-tbfed-pagefooter/footer.css">
                
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-katex/katex.min.css">
                
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-hints/plugin-hints.css">
                
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-tabs/bootstrap.css">
                
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-lightbox/css/lightbox.min.css">
                
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-search-plus/search.css">
                
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-highlight/website.css">
                
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-fontsettings/website.css">
                
            
        

    

    

        
    
    
    <meta name="HandheldFriendly" content="true">
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <link rel="apple-touch-icon-precomposed" sizes="152x152" href="../../gitbook/images/apple-touch-icon-precomposed-152.png">
    <link rel="shortcut icon" href="../../gitbook/images/favicon.ico" type="image/x-icon">

    
    <link rel="next" href="bam.html">
    
    
    <link rel="prev" href="realign-outputs.html">
    

    </head>
    <body>
        
<div class="book">
    <div class="book-summary"><div class="book-logo-root"><div class="book-logo" style="text-align: center; padding: 20px;"><img src="data:image/png;base64,AAAAGGZ0eXBhdmlmAAAAAG1pZjFtaWFmAAAA0m1ldGEAAAAAAAAAIWhkbHIAAAAAAAAAAHBpY3QAAAAAAAAAAAAAAAAAAAAADnBpdG0AAAAAAAEAAAAeaWxvYwAAAABEAAABAAEAAAABAAAA8gAABgAAAAAjaWluZgAAAAAAAQAAABVpbmZlAgAAAAABAABhdjAxAAAAAFZpcHJwAAAAOGlwY28AAAAUaXNwZQAAAAAAAAAgAAAALQAAAAxhdjFDgT9AAAAAABBwaXhpAAAAAAMKCgoAAAAWaXBtYQAAAAAAAAABAAEDAYIDAAAGCG1kYXQSAAoJP9F/YXgIaDaAMvALZAQYAAQAAgIAAAAAAAAAEAAAAAVACJZH5jEfcnHq5GfVOqfqAcroK/CW/n75YIMe8ZfAjZNFE/6UrpMmpDP/er+E0q/ceflnZGp4dyYbk4GCBlbhdKT3x+D/AIaisuTFd8nYmo+ez1m+B7mjO9pucuLd4J09fZofEcRZ1N8yXb1uByDJRlxGRNCsg51xqZYiGzKHoIgR4ChwET50S4FSgHUScr3K3YmkXZjZK1Sle8EZjq6uDLYWGw4NvJ/pvNMkd/WmxjuUTT7pfVtNIGjG28GzYzHlf3vaTaGjRUC91DO9vrJT51k9bZZg8h2LiVbz7yNTOZcjlBLZuokz15s00uLVyWobV2CGySmbuxBbAsKxItjkVbSHo6ojiJbY5bPJvksg0SaezQgA1EgcN96PcNp/fjOFRBtevV/O+rjq23jHY8C9n/YSCdXnGsWJNHdxleLb40bjcNXoXncQADGMuX4XOfKPzQ3L6GO6nlgQBXZyDdvAAbF778o6nSAV86cHKEWnD3tckPflzq+6jvx4oCjObYt0X1Ro63jOf55vCFAlxB8FJNtvuok/MW0bSKvp+0LTGESD/2lmWnH+IvOX5hbFmtTw5mgGW0eKFHyZXqg00hWU4wfzKPxpRwvxAUZMaiFlEudhK9HjhFYdgz/1ShJlEUXzbR+U6nabVjctZv4ZQeECNYZ3p4pAGxtsSlxZBYQ4ARr8zqcDQjsYk+rr1KdIQ14PnyCIe0imW/rJ/at5+YUvZ6Z4FApMELY7udGJJ5HuVaPCr22ij/h9mvhjgSMMfCesXRgyxDAWg9Aa+lx/LxDWkXLxHot/ByjPyng7o28F0TwpKuvCU8x8MxSuy9o89zSiGVR2nsz7EEWJvFEeZKRMn9j30VIwqcF4D/tWMO3puIoih0v/CvKusdf7v0qTb58ZaIGvRrTJSLiosclEEtiORGa0wihqVOcL7O3GL2rIRP/Pcv1BSDYXhybnLZ+HbhjuBO2ulzWv3j+0GWtGdtmij8t9ivPRKOC7RYXLQ99mYd621NSxNWbNJaP1XYbLFkbATt+nBDr+ypqyy+ARzVlE9ywft4Zj683T2knXPPhbo0X9I70mu1LIAsFCr9IwunPhTV//P4W4w8S1Ks9IB7NeBdiWCCtW1KWtIuFaIPvx+VbZkxyad+Nqh4iFTKBYIfGffIbiEQiBaJaxB+uIC8/TylgtlADj00ipv0HgSHv3v2rna/hOViWRbA9kars4YRbv8GF4i4delTz8Myp9DsZQv/owkRFJkk5KHrv25DT0CrW2FxlBIuJQuSHak+OCRtl+RnJwNd+0wDZ09WRFCkbt6qoCbVumBWzBM/0afWjK8QukftmXeX93+lbWGWrOova/TResZhzunNubiTDOH6WgLh/g8v3YUg1S5V2GzESjFUtpQBZFiQVn6a+beAxr3kQINzRSASF1nZZjdiEmN7rXfuhoqkBJV6uARAJBDfPNARGsjTgivGjqjHdjeZHNowz9apljGORFaJJBbE0zTjcd7NUmAiSd9dIXG8cAmiBmmU1fUbaupjLvmwgxdI9ZxfpP3h7WygcJklKiA+Unx+LJT20S7nW1nO/HiqEKOAnhKKEvpWo8NWVVJ1l1HXanmZRnuA+DkpOr4ahp63B1LxmbWnEP8IzsZtbQWG1+yQUfwN4CYnnPZB34ljqnPbGb+ReVnQ/B552w6HkZFlnSYxvuVvAg6LnzslXN9UZZreNj6xTh9AAsznkSmpb7//5b4QpSnH9N948K/lP3QND+g6Co5XKGmAfUS+NnBs4/AzZQg+3n/VjAIHA5YIMMduPmXmlwzDiQOZC2pSYfpMUqIScTuITS35U5C8lq86m1jGD1ZSZRse4fJdFbPKs+TldyLZyqXub516/e1dRG47gRwSUoWfcjoG486GAZTabIiCOxD9Pq13HejOuZG+j+/0zALlXdSctzuOMx1XJMaYHpJ06kZmkCGN+JXeuDjk32LVGFUn/WY7XCw1tvJQBykA6dQTtjhbrhALglaNHrFiw=" style><p class="book-title" style="font-weight:600">SAW &#x7528;&#x6237;&#x624B;&#x518C; 8.1 </p></div></div>
        
            
<div id="book-search-input" role="search">
    <input type="text" placeholder="&#x8F93;&#x5165;&#x5E76;&#x641C;&#x7D22;">
</div>

            
                <nav role="navigation">
                


<ul class="summary">
    
    

    

    
        
        
    
        <li class="chapter " data-level="1.1" data-path="../../">
            
                <a href="../../">
            
                    
                    &#x6982;&#x89C8;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.2" data-path="../../getting-started/">
            
                <a href="../../getting-started/">
            
                    
                    &#x5FEB;&#x901F;&#x5F00;&#x59CB;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.2.1" data-path="../../getting-started/introduction.html">
            
                <a href="../../getting-started/introduction.html">
            
                    
                    &#x8F6F;&#x4EF6;&#x4ECB;&#x7ECD;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.2.2" data-path="../../getting-started/glossary.html">
            
                <a href="../../getting-started/glossary.html">
            
                    
                    &#x8BCD;&#x6C47;&#x8868;
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.3" data-path="../../download-center.html">
            
                <a href="../../download-center.html">
            
                    
                    &#x4E0B;&#x8F7D;&#x4E2D;&#x5FC3;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4" data-path="../">
            
                <a href="../">
            
                    
                    &#x8FD0;&#x884C;&#x5206;&#x6790;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.1" data-path="../inputs/">
            
                <a href="../inputs/">
            
                    
                    &#x8F93;&#x5165;&#x6587;&#x4EF6;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.1.1" data-path="../inputs/chip-mask.html">
            
                <a href="../inputs/chip-mask.html">
            
                    
                    &#x82AF;&#x7247; mask
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.1.2" data-path="../inputs/fastqs.html">
            
                <a href="../inputs/fastqs.html">
            
                    
                    &#x6D4B;&#x5E8F; FASTQ
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.1.3" data-path="../inputs/images.html">
            
                <a href="../inputs/images.html">
            
                    
                    &#x663E;&#x5FAE;&#x955C;&#x56FE;&#x50CF;
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.4.2" data-path="../pipelines/">
            
                <a href="../pipelines/">
            
                    
                    &#x5206;&#x6790;&#x6D41;&#x7A0B;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.2.1" data-path="../pipelines/saw-can-shu-ming-ling.html">
            
                <a href="../pipelines/saw-can-shu-ming-ling.html">
            
                    
                    SAW &#x53C2;&#x6570;&#x547D;&#x4EE4;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.2.2" data-path="../pipelines/count/">
            
                <a href="../pipelines/count/">
            
                    
                    count
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.2.2.1" data-path="../pipelines/count/stereo-seq-t-ff.html">
            
                <a href="../pipelines/count/stereo-seq-t-ff.html">
            
                    
                    Stereo-seq T FF
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.2.2.2" data-path="../pipelines/count/stereo-seq-n-ffpe-v1.0.html">
            
                <a href="../pipelines/count/stereo-seq-n-ffpe-v1.0.html">
            
                    
                    Stereo-seq N FFPE
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.2.2.3" data-path="../pipelines/count/stereo-cite-t-ff.html">
            
                <a href="../pipelines/count/stereo-cite-t-ff.html">
            
                    
                    Stereo-CITE T FF
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.4.3" data-path="./">
            
                <a href="./">
            
                    
                    &#x8F93;&#x51FA;&#x7ED3;&#x679C;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.3.1" data-path="count-outputs.html">
            
                <a href="count-outputs.html">
            
                    
                    count &#x8F93;&#x51FA;&#x7ED3;&#x679C;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.2" data-path="realign-outputs.html">
            
                <a href="realign-outputs.html">
            
                    
                    realign &#x8F93;&#x51FA;&#x7ED3;&#x679C;
            
                </a>
            

            
        </li>
    
        <li class="chapter active" data-level="1.4.3.3" data-path="html-report.html">
            
                <a href="html-report.html">
            
                    
                    HTML report
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.4" data-path="bam.html">
            
                <a href="bam.html">
            
                    
                    BAM
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.5" data-path="matrices.html">
            
                <a href="matrices.html">
            
                    
                    Matrices
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.6" data-path="images.html">
            
                <a href="images.html">
            
                    
                    Images
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.7" data-path="analysis.html">
            
                <a href="analysis.html">
            
                    
                    Analysis
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.5" data-path="../../shi-yong-jiao-cheng/">
            
                <a href="../../shi-yong-jiao-cheng/">
            
                    
                    &#x4F7F;&#x7528;&#x6559;&#x7A0B;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.5.1" data-path="../../shi-yong-jiao-cheng/preparation-of-reference.html">
            
                <a href="../../shi-yong-jiao-cheng/preparation-of-reference.html">
            
                    
                    &#x6BD4;&#x5BF9;&#x7D22;&#x5F15;&#x6784;&#x5EFA;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.2" data-path="../../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/">
            
                <a href="../../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/">
            
                    
                    &#x8FD0;&#x884C;&#x6838;&#x5FC3;&#x5206;&#x6790;&#x6D41;&#x7A0B;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.5.2.1" data-path="../../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/stereo-seq-ff.html">
            
                <a href="../../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/stereo-seq-ff.html">
            
                    
                    Stereo-seq FF &#x5206;&#x6790;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.2.2" data-path="../../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/stereo-seq-ffpe.html">
            
                <a href="../../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/stereo-seq-ffpe.html">
            
                    
                    Stereo-seq FFPE &#x5206;&#x6790;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.2.3" data-path="../../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/stereocite-ff-fen-xi.html">
            
                <a href="../../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/stereocite-ff-fen-xi.html">
            
                    
                    Stereo-CITE FF &#x5206;&#x6790;
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.5.3" data-path="../../shi-yong-jiao-cheng/shou-dong-tu-xiang-chu-li.html">
            
                <a href="../../shi-yong-jiao-cheng/shou-dong-tu-xiang-chu-li.html">
            
                    
                    &#x624B;&#x52A8;&#x56FE;&#x50CF;&#x5904;&#x7406;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.4" data-path="../../shi-yong-jiao-cheng/secondary-analysis.html">
            
                <a href="../../shi-yong-jiao-cheng/secondary-analysis.html">
            
                    
                    &#x6570;&#x636E;&#x518D;&#x5206;&#x6790;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.5" data-path="../../shi-yong-jiao-cheng/shu-ju-ge-shi-zhuan-huan.html">
            
                <a href="../../shi-yong-jiao-cheng/shu-ju-ge-shi-zhuan-huan.html">
            
                    
                    &#x6570;&#x636E;&#x683C;&#x5F0F;&#x8F6C;&#x6362;
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.6" data-path="../../algorithms/">
            
                <a href="../../algorithms/">
            
                    
                    &#x7B97;&#x6CD5;&#x539F;&#x7406;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.6.1" data-path="../../algorithms/gene-expression-algorithms.html">
            
                <a href="../../algorithms/gene-expression-algorithms.html">
            
                    
                    Read processing algorithms
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.6.2" data-path="../../algorithms/image-processing-algorithms.html">
            
                <a href="../../algorithms/image-processing-algorithms.html">
            
                    
                    Image processing algorithms
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.7" data-path="../../gao-ji-she-zhi/">
            
                <a href="../../gao-ji-she-zhi/">
            
                    
                    &#x9AD8;&#x7EA7;&#x8BBE;&#x7F6E;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.7.1" data-path="../../gao-ji-she-zhi/analysis-with-gpu.html">
            
                <a href="../../gao-ji-she-zhi/analysis-with-gpu.html">
            
                    
                    Analysis with GPU
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.7.2" data-path="../../gao-ji-she-zhi/expression-matrix-format.html">
            
                <a href="../../gao-ji-she-zhi/expression-matrix-format.html">
            
                    
                    Expression matrix format
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.7.3" data-path="../../gao-ji-she-zhi/image-related-record-format.html">
            
                <a href="../../gao-ji-she-zhi/image-related-record-format.html">
            
                    
                    Image-related record format
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.8" data-path="../../release-notes.html">
            
                <a href="../../release-notes.html">
            
                    
                    &#x7248;&#x672C;&#x8BF4;&#x660E;
            
                </a>
            

            
        </li>
    

    
        
        <li class="header">&#x6280;&#x672F;&#x652F;&#x6301;</li>
        
        
    
        <li class="chapter " data-level="2.1" data-path="../../support/shu-ju-ji.html">
            
                <a href="../../support/shu-ju-ji.html">
            
                    
                    &#x6570;&#x636E;&#x96C6;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="2.2" data-path="../../support/cuo-wu-ma.html">
            
                <a href="../../support/cuo-wu-ma.html">
            
                    
                    &#x9519;&#x8BEF;&#x7801;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="2.3" data-path="../../support/shi-pei-jian-rong-de-ruan-jian-he-shi-ji-he.html">
            
                <a href="../../support/shi-pei-jian-rong-de-ruan-jian-he-shi-ji-he.html">
            
                    
                    &#x9002;&#x914D;&#x517C;&#x5BB9;&#x7684;&#x8F6F;&#x4EF6;&#x548C;&#x8BD5;&#x5242;&#x76D2;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="2.4">
            
                <a target="_blank" href="https://www.stomics.tech/resources/FAQ">
            
                    
                    &#x5E38;&#x89C1;&#x95EE;&#x9898;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="2.5">
            
                <a target="_blank" href="https://www.stomics.tech/service/new-StereoMap.html">
            
                    
                    StereoMap
            
                </a>
            

            
        </li>
    

    

    <li class="divider"></li>

    <li>
        <a href="https://www.gitbook.com" target="blank" class="gitbook-link">
            &#x672C;&#x4E66;&#x4F7F;&#x7528; GitBook &#x53D1;&#x5E03;
        </a>
    </li>
</ul>


                </nav>
            
        
    </div>

    <div class="book-body">
        
            <div class="body-inner">
                
                    

<div class="book-header" role="navigation">
    

    <!-- Title -->
    <h1>
        <i class="fa fa-circle-o-notch fa-spin"></i>
        <a href="../..">HTML report</a>
    </h1>
</div>




                    <div class="page-wrapper" tabindex="-1" role="main">
                        <div class="page-inner">
                            
<div class="search-plus" id="book-search-results">
    <div class="search-noresults">
    
                                <section class="normal markdown-section">
                                
                                <div id="anchor-navigation-ex-navbar"><i class="fa fa-navicon"></i><ul><li><span class="title-icon "></span><a href="#html-report"><b></b>HTML report</a></li><ul><li><span class="title-icon "></span><a href="#summary"><b></b>Summary</a></li><ul><li><span class="title-icon "></span><a href="#key-metrics"><b></b>Key metrics</a></li><li><span class="title-icon "></span><a href="#annotation"><b></b>Annotation</a></li><li><span class="title-icon "></span><a href="#tissue-related"><b></b>Tissue related</a></li><li><span class="title-icon "></span><a href="#sequencing-saturation"><b></b>Sequencing saturation</a></li><li><span class="title-icon "></span><a href="#information"><b></b>Information</a></li></ul><li><span class="title-icon "></span><a href="#square-bin"><b></b>Square Bin</a></li><ul><li><span class="title-icon "></span><a href="#statistics"><b></b>Statistics</a></li><li><span class="title-icon "></span><a href="#plots"><b></b>Plots</a></li><li><span class="title-icon "></span><a href="#clustering--umap"><b></b>Clustering &amp; UMAP</a></li><li><span class="title-icon "></span><a href="#differential-expression-analysis"><b></b>Differential expression analysis</a></li></ul><li><span class="title-icon "></span><a href="#cell-bin"><b></b>Cell Bin</a></li><ul><li><span class="title-icon "></span><a href="#statistics_1"><b></b>Statistics</a></li><li><span class="title-icon "></span><a href="#plots_1"><b></b>Plots</a></li><li><span class="title-icon "></span><a href="#clustering--umap_1"><b></b>Clustering &amp; UMAP</a></li><li><span class="title-icon "></span><a href="#differential-expression-analysis_1"><b></b>Differential expression analysis</a></li><li><span class="title-icon "></span><a href="#image-information"><b></b>Image information</a></li><li><span class="title-icon "></span><a href="#qc"><b></b>QC</a></li><li><span class="title-icon "></span><a href="#stitching"><b></b>Stitching</a></li><li><span class="title-icon "></span><a href="#registration"><b></b>Registration</a></li></ul><li><span class="title-icon "></span><a href="#microorganism"><b></b>Microorganism</a></li><ul><li><span class="title-icon "></span><a href="#denoising"><b></b>Denoising</a></li><li><span class="title-icon "></span><a href="#taxonomic-classification"><b></b>Taxonomic Classification</a></li><li><span class="title-icon "></span><a href="#microbes-proportion-phylum"><b></b>Microbes Proportion (Phylum)</a></li></ul><li><span class="title-icon "></span><a href="#summary-protein"><b></b>Summary-Protein</a></li><ul><li><span class="title-icon "></span><a href="#key-metrics_1"><b></b>Key metrics</a></li><li><span class="title-icon "></span><a href="#sequencing-saturation_1"><b></b>Sequencing saturation</a></li><li><span class="title-icon "></span><a href="#protein-correlations"><b></b>Protein correlations</a></li><li><span class="title-icon "></span><a href="#gene--protein-correlations"><b></b>Gene : protein correlations</a></li><li><span class="title-icon "></span><a href="#histogram-of-portein-counts"><b></b>Histogram of portein counts</a></li><li><span class="title-icon "></span><a href="#information_1"><b></b>Information</a></li></ul><li><span class="title-icon "></span><a href="#square-bin-protein"><b></b>Square Bin-Protein</a></li><ul><li><span class="title-icon "></span><a href="#statistics_2"><b></b>Statistics</a></li><li><span class="title-icon "></span><a href="#plots_2"><b></b>Plots</a></li><li><span class="title-icon "></span><a href="#clustering--umap_2"><b></b>Clustering &amp; UMAP</a></li></ul><li><span class="title-icon "></span><a href="#cell-bin-protein"><b></b>Cell Bin-Protein</a></li><ul><li><span class="title-icon "></span><a href="#statistics_3"><b></b>Statistics</a></li><li><span class="title-icon "></span><a href="#plots_3"><b></b>Plots</a></li><li><span class="title-icon "></span><a href="#clustering--umap_3"><b></b>Clustering &amp; UMAP</a></li></ul><li><span class="title-icon "></span><a href="#analysis-no-longer-available-since-812"><b></b>Analysis (no longer available since 8.1.2)</a></li><ul><li><span class="title-icon "></span><a href="#multiomics-clustering--umap"><b></b>Multiomics clustering &amp; UMAP</a></li><li><span class="title-icon "></span><a href="#top-markers-by-cluster"><b></b>Top markers by cluster</a></li></ul><li><span class="title-icon "></span><a href="#alerts"><b></b>Alerts</a></li></ul></ul></div><a href="#html-report" id="anchorNavigationExGoTop"><i class="fa fa-arrow-up"></i></a><h1 id="html-report"><a name="html-report" class="anchor-navigation-ex-anchor" href="#html-report"><i class="fa fa-link" aria-hidden="true"></i></a>HTML report</h1>
<p><em><mark style="color:purple;">*The interpretation of the HTML report is only available in EN.</mark></em></p>
<p><code>SAW count</code> and <code>SAW realign</code> pipelines will output an interactive report <code>&lt;SN&gt;.report.html</code>. The contents of the HTML report file will vary depending on the pipeline and parameters used but generally follow a similar format across runs.</p>
<p>On this page, we demonstrate the report of </p>
<ul>
<li>a mouse brain sample from Stereo-seq T FF V1.3,</li>
<li>a mouse lung tissue sample from  Stereo-seq N FFPE V1.0,</li>
<li>and a mouse thymus tissue sample from Stereo-CITE T V1.0.</li>
</ul>
<p>Run with  SAW v8.1.</p>
<h2 id="summary"><a name="summary" class="anchor-navigation-ex-anchor" href="#summary"><i class="fa fa-link" aria-hidden="true"></i></a><strong>Summary</strong></h2>
<figure><a href="../../img/assets/summary-exprssion.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/summary-exprssion.PNG" alt></a><figcaption><p>Expression heatmap and four key metrics</p></figcaption></figure>

<figure><a href="../../img/assets/summary-image.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/summary-image.PNG" alt></a><figcaption><p>Display of microscope image</p></figcaption></figure>

<p>The spatial gene expression distribution plot, containing all bins and bins under tissue, on the left, shows MID count at each bin20.</p>
<p><code>Total Reads</code> is the amount of total sequencing reads of input FASTQs. <code>Mean MID per Bin20/Mean MID per Bin50</code> and <code>Mean Gene per Bin20/Mean Gene per Bin50</code> represent the mean MID and gene type counts at each bin20 or bin50. <code>Total Genes</code> is the number of gene types from all bins.</p>
<h3 id="key-metrics"><a name="key-metrics" class="anchor-navigation-ex-anchor" href="#key-metrics"><i class="fa fa-link" aria-hidden="true"></i></a>Key metrics</h3>
<figure><a href="../../img/assets/summary-key-metrics.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/summary-key-metrics.PNG" alt></a><figcaption><p>Details and sunburst plot of key metrics</p></figcaption></figure>

<p>Key metrics of the data are listed:</p>
<table><thead><tr><th width="227">Metric</th><th>Description</th></tr></thead><tbody><tr><td>Total Reads</td><td>Total number of sequenced reads.</td></tr><tr><td><a href="../../algorithms/gene-expression-algorithms.html#cid-mapping">Valid CID Reads</a></td><td>Number of reads with CIDs that can be matched with the mask file</td></tr><tr><td>Invalid CID Reads</td><td>Number of reads with CIDs that cannot be matched with the mask file.</td></tr><tr><td>Clean Reads</td><td>Number of Valid CID Reads that have passed QC.</td></tr><tr><td><a href="../../algorithms/gene-expression-algorithms.html#rna-filtering">Non-Relevant Short Reads</a></td><td>Number of non-relevant short reads</td></tr><tr><td>Discarded MID Reads</td><td>Number of reads with MID that have been discarded since MID sequence quality does not satisfy with further analysis.</td></tr><tr><td>Uniquely Mapped Reads</td><td>Number of reads that mapped uniquely to the reference genome. If the pipeline uses uniquely mapped reads and the best match from multi-mapped reads for subsequent annotation, this item will include them both</td></tr><tr><td><a href="../../algorithms/gene-expression-algorithms.html#annotation-to-transcriptome">Transcriptome</a></td><td>Number of reads that are aligned to transcripts of at least one gene</td></tr><tr><td>Unique Reads</td><td>Number of reads in Transcriptome that have been corrected by MAPQ and deduplicated</td></tr><tr><td>Sequencing Saturation</td><td>Number of reads in Transcriptome that have been corrected by MAPQ with duplicated MID</td></tr><tr><td>Unannotated Reads</td><td>Number of reads that cannot be aligned to the transcript of one gene</td></tr><tr><td>Multi-Mapped Reads</td><td>Number of reads that mapped more than one time on the genome. If the pipeline uses uniquely mapped reads and the best match from multi-mapped reads for subsequent annotation, this item will exclude multi-mapped ones to be annotated</td></tr><tr><td>Unmapped Reads</td><td>Number of reads that cannot be mapped to the reference genome.</td></tr><tr><td>rRNA Reads</td><td>Number of reads that mapped to the rRNA regions.</td></tr></tbody></table>

<h3 id="annotation"><a name="annotation" class="anchor-navigation-ex-anchor" href="#annotation"><i class="fa fa-link" aria-hidden="true"></i></a>Annotation</h3>
<p>Metrics of reads to be annotated by GTF/GFF files.</p>
<table><thead><tr><th width="234">Metric</th><th>Description</th></tr></thead><tbody><tr><td>Exonic</td><td>Number of reads that mapped uniquely to an exonic region and on the same strand of the genome.</td></tr><tr><td>Intronic</td><td>Number of reads that mapped uniquely to an intronic region and on the same strand of the genome.</td></tr><tr><td>Intergenic</td><td>Number of reads that mapped uniquely to an intergenic region and on the same strand of the genome.</td></tr><tr><td>Antisense</td><td>Number of reads mapped to the transcriptome but on the opposite strand of their annotated gene.</td></tr></tbody></table>



<h3 id="tissue-related"><a name="tissue-related" class="anchor-navigation-ex-anchor" href="#tissue-related"><i class="fa fa-link" aria-hidden="true"></i></a>Tissue related</h3>
<figure><a href="../../img/assets/summary-tissue-segmentation.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/summary-tissue-segmentation.PNG" alt></a><figcaption><p>Display and metrics related to tissue-coverage region</p></figcaption></figure>

<p>The tissue segmentation result based on a microscope image is shown on the left, of which the tissue region is covered in purple.</p>
<p>Metrics related to tissue coverage are listed:</p>
<table><thead><tr><th width="244">Metric</th><th>Description</th></tr></thead><tbody><tr><td>Tissue Area</td><td>Tissue area in mm&#xB2;&#x200B;.</td></tr><tr><td>Number of MID Under Tissue Coverage</td><td>Number of MID under tissue coverage.</td></tr><tr><td>Fraction MID in Spots Under Tissue</td><td><p>Fraction of MID under tissue over total unique reads.</p><p> (MID Under Tissue / Unique Reads)</p></td></tr></tbody></table>

<h3 id="sequencing-saturation"><a name="sequencing-saturation" class="anchor-navigation-ex-anchor" href="#sequencing-saturation"><i class="fa fa-link" aria-hidden="true"></i></a>Sequencing saturation</h3>
<figure><a href="../../img/assets/summary-saturation%20(1).PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/summary-saturation (1).PNG" alt></a><figcaption><p>Sequencing saturation curves</p></figcaption></figure>

<p>The saturation analysis in the HTML report can assess the overall quality of the sequencing data. In order to improve calculation efficiency, small samples are randomly selected from successfully annotated reads in the bin20 or bin50 dimension. Therefore, the results of multiple runs of the same data may vary slightly. The formulas may not be identical, but the general shape of the curve is consistent.</p>
<ul>
<li>Figure 1: As the number of random samples increases, the gene median in the bin20 dimension gradually increases.</li>
<li>Figure 2: Curves fitted based on Unique Reads data from randomly sampled samples.</li>
<li><strong>Figure 3</strong>: Statistics of Unique Reads (reads with unique CID, geneName and MID) in the sampled samples, saturation value = 1-(Unique Reads)/(Total Annotated Reads), as the sampling volume increases, the fitting curve becomes near-flat, indicating that the data tends to be saturated. Whether to add additional tests depends on the overall project design and sample conditions. For example, it is recommended that additional tests be performed on precious samples. The threshold value of 0.8 in the report serves as a reminder for recommended guidance.</li>
</ul>
<p></p><div class="alert alert-info hints-alert"><div class="hints-icon"><i class="fa fa-info-circle"></i></div><div class="hints-container"><p>The x-axis of the three graphs is the same, and the y-axis is divided into saturation value, gene median, and number of Unique Reads.</p>
</div></div><p></p>
<h3 id="information"><a name="information" class="anchor-navigation-ex-anchor" href="#information"><i class="fa fa-link" aria-hidden="true"></i></a>Information</h3>
<p>This item displays the basic information of the input FASTQs, </p>
<p><code>Species</code> is from the <code>--organism</code> parameter used in <code>SAW count</code>, usually referring to the species.</p>
<p><code>Tissue</code> is from the <code>--tissue</code> parameter used in <code>SAW count</code>.</p>
<p><code>Reference</code> means the reference genome used in <code>SAW count</code>, as the same as <code>Organism</code>.</p>
<p><code>FASTQ</code> records FASTQ files in <code>SAW count</code>, including file prefixes of all input sequencing FASTQs.</p>
<h2 id="square-bin"><a name="square-bin" class="anchor-navigation-ex-anchor" href="#square-bin"><i class="fa fa-link" aria-hidden="true"></i></a><strong>Square Bin</strong></h2>
<p>This page contains results of statistics, plots, clustering, UMAP, and differential expression analysis, at bin dimension. Results come from the analysis based on <code>&lt;SN&gt;.tissue.gef</code> file.</p>
<h3 id="statistics"><a name="statistics" class="anchor-navigation-ex-anchor" href="#statistics"><i class="fa fa-link" aria-hidden="true"></i></a>Statistics</h3>
<figure><a href="../../img/assets/squarebin-table.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/squarebin-table.PNG" alt></a><figcaption><p>Statistics of bins under tissue-coverage region</p></figcaption></figure>

<p>The above table records the statistics is bin20, bin50 and bin200:</p>
<table><thead><tr><th width="247">Item</th><th>Description</th></tr></thead><tbody><tr><td>Bin Size</td><td><p>The size of Bin which is the unit of aggregated DNBs in a squared region.</p><p>i.e. Bin 50 = 50 * 50 DNBs</p></td></tr><tr><td>Mean Reads (per bin)</td><td>Mean number of sequenced reads divided by the number of bins under tissue coverage.</td></tr><tr><td>Median Reads (per bin)</td><td>Median number of sequenced reads divided by the number of bins under tissue coverage (pick the middle value after sorting).</td></tr><tr><td>Mean Gene Type (per bin)</td><td>Mean number of unique gene types divided by the number of bins under tissue coverage.</td></tr><tr><td>Median Gene Type (per bin)</td><td>Median number of unique gene types divided by the number of bins under tissue coverage.</td></tr><tr><td>Mean MID (per bin)</td><td>Mean number of MIDs divided by the number of bins under tissue coverage.</td></tr><tr><td>Median MID (per bin)</td><td>Median number of MIDs divided by the number of bins under tissue coverage</td></tr></tbody></table>

<h3 id="plots"><a name="plots" class="anchor-navigation-ex-anchor" href="#plots"><i class="fa fa-link" aria-hidden="true"></i></a>Plots</h3>
<figure><a href="../../img/assets/squarebin-violin.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/squarebin-violin.PNG" alt></a><figcaption><p>Distribution plots of MID and gene type</p></figcaption></figure>

<p>Violin plots show the distribution of deduplicated MID count and gene types in each bin.</p>
<h3 id="clustering--umap"><a name="clustering--umap" class="anchor-navigation-ex-anchor" href="#clustering--umap"><i class="fa fa-link" aria-hidden="true"></i></a><strong>Clustering &amp; UMAP</strong></h3>
<figure><a href="../../img/assets/squarebin-cluster.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/squarebin-cluster.PNG" alt></a><figcaption><p>Leiden clustering and UMAP projection</p></figcaption></figure>

<p>Clustering is performed based on <code>SN.tissue.gef</code> using the Leiden algorithm. UMAP projections are performed based on <code>SN.tissue.gef</code> and colored by automated clustering. The same color is assigned to spots that are within a shorter distance and with similar gene expression profiles.</p>
<h3 id="differential-expression-analysis"><a name="differential-expression-analysis" class="anchor-navigation-ex-anchor" href="#differential-expression-analysis"><i class="fa fa-link" aria-hidden="true"></i></a>Differential expression analysis</h3>
<figure><a href="../../img/assets/squarebin-df-table.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/squarebin-df-table.PNG" alt></a><figcaption><p>Marker feature table</p></figcaption></figure>

<p>The goal of the differential expression analysis is to identify markers that are more highly expressed in a cluster than the rest of the sample. For each marker, a differential expression test was run between each cluster and the remaining sample. An estimate of the log2 ratio of expression in a cluster to that in other coordinates is Log2 fold-change (L2FC). A value of 1.0 denotes a 2-fold increase in expression within the relevant cluster. Based on a negative binomial test, the p-value indicates the expression difference&apos;s statistical significance. The Benjamini-Hochberg method has been used to correct the p-value for multiple testing. Additionally, the top N features by L2FC for each cluster were kept after features in this table were filtered by (Mean UMI counts &gt; 1.0). Grayed-out features have an adjusted p-value &gt;= 0.10 or an L2FC &lt; 0. N (ranges from 1 to 50) is the number of top features displayed per cluster, which is set to limit the amount of table entries displayed to 10,000. N=%10,000/K^2 where K is the number of clusters. Click on a column to sort by that value, or search a gene of interest.</p>
<p></p><div class="alert alert-info hints-alert"><div class="hints-icon"><i class="fa fa-info-circle"></i></div><div class="hints-container"><p>When the values of L2FC in the marker feature table are blank, &quot;infinity&quot; and &quot;-infinity&quot;, the analysis results are normal. These conditions are well explained below.</p>
</div></div><p></p>
<p>The calculation of L2FC is related to the expression number of cells of a certain gene in the case group and the control group. Since the calculation of L2FC uses the natural logarithm as the base, when the expression relationship has extremely high or low values, the three special values, none, &quot;inf&quot; and &quot;-inf&quot;, will appear. The screenshot below uses inf and a constant to make a simple demonstration.</p>
<figure><a href="../../img/assets/image%20(6).png" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/image (6).png" alt></a><figcaption><p>An example in Notebook using Python</p></figcaption></figure>

<p></p><div class="alert alert-undefined hints-alert"><div class="hints-icon"><i class="undefined"></i></div><div class="hints-container"><p>The p-values should be increasing as the list descends (with a maximum of 1), infinitely close to 0.</p>
<p>If you find that the p-value is 0 in the result table, it may be because the calculated differential expression feature is extremely significant, leading to an extremely small p-value. This can exceed the limit of the data type (usually <code>float64</code>, depending on the basic computing package), resulting in a situation that cannot be expressed in scientific notation.</p>
</div></div><p></p>
<h2 id="cell-bin"><a name="cell-bin" class="anchor-navigation-ex-anchor" href="#cell-bin"><i class="fa fa-link" aria-hidden="true"></i></a><strong>Cell Bin</strong></h2>
<p>This page contains results of statistics, plots, clustering, UMAP, and differential expression analysis, at cellbin dimension. Cell border expanding is automatically performed during <code>SAW count</code> and <code>SAW realign</code>, which means the contents of &quot;Cell Bin&quot; tab are based on <code>SN.adjusted.cellbin.gef</code>. </p>
<p></p><div class="alert alert-undefined hints-alert"><div class="hints-icon"><i class="undefined"></i></div><div class="hints-container"><p>When it comes to <code>--adjusted-distance=0</code> in <code>SAW realign</code>, all contents of this tab are based on <code>SN.cellbin.gef</code>. </p>
</div></div><p></p>
<h3 id="statistics_1"><a name="statistics_1" class="anchor-navigation-ex-anchor" href="#statistics_1"><i class="fa fa-link" aria-hidden="true"></i></a>Statistics</h3>
<figure><a href="../../img/assets/cellbin-table%20(1).PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/cellbin-table (1).PNG" alt></a><figcaption><p>Detailed statistics of cellbin</p></figcaption></figure>

<p>The above table records the statistics of cellbin:</p>
<table><thead><tr><th width="298">Item</th><th>Description</th></tr></thead><tbody><tr><td>Cell Count</td><td>Number of cells.</td></tr><tr><td>Mean Cell Area</td><td>Mean cell area, in pixes.</td></tr><tr><td>Median Cell Area</td><td>Median cell area, in pixes.</td></tr><tr><td>Mean Gene Type</td><td>Mean gene types per cell.</td></tr><tr><td>Median Gene Type</td><td>Median gene types per cell.</td></tr><tr><td>Mean MID</td><td>Mean MID count per cell.</td></tr><tr><td>Median MID</td><td>Median MID count per cell.</td></tr></tbody></table>

<h3 id="plots_1"><a name="plots_1" class="anchor-navigation-ex-anchor" href="#plots_1"><i class="fa fa-link" aria-hidden="true"></i></a>Plots</h3>
<figure><a href="../../img/assets/cellbin-plot.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/cellbin-plot.PNG" alt></a><figcaption><p>Distribution plots of MID and gene type</p></figcaption></figure>

<p>Violin plots show the distribution of deduplicated MID count, gene types and cell area in the cellbin.</p>
<h3 id="clustering--umap_1"><a name="clustering--umap_1" class="anchor-navigation-ex-anchor" href="#clustering--umap_1"><i class="fa fa-link" aria-hidden="true"></i></a><strong>Clustering &amp; UMAP</strong></h3>
<figure><a href="../../img/assets/cellbin-cluster.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/cellbin-cluster.PNG" alt></a><figcaption><p>Leiden clustering and UMAP projection</p></figcaption></figure>

<p>Clustering is performed based on <code>SN.adjusted.cellbin.gef</code> or <code>SN.cellbin.gef</code>, using the Leiden algorithm. UMAP projections are performed based on <code>SN.adjusted.cellbin.gef</code> or <code>SN.cellbin.gef</code>, and colored by automated clustering. The same color is assigned to spots that are within a shorter distance and with similar gene expression profiles.</p>
<h3 id="differential-expression-analysis_1"><a name="differential-expression-analysis_1" class="anchor-navigation-ex-anchor" href="#differential-expression-analysis_1"><i class="fa fa-link" aria-hidden="true"></i></a>Differential expression analysis</h3>
<figure><a href="../../img/assets/cellbin-df-table%20(1).PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/cellbin-df-table (1).PNG" alt></a><figcaption><p>Marker feature table</p></figcaption></figure>

<p>The goal of the differential expression analysis is to identify markers that are more highly expressed in a cluster than the rest of the sample. For each marker, a differential expression test was run between each cluster and the remaining sample. An estimate of the log2 ratio of expression in a cluster to that in other coordinates is Log2 fold-change (L2FC). A value of 1.0 denotes a 2-fold increase in expression within the relevant cluster. Based on a negative binomial test, the p-value indicates the expression difference&apos;s statistical significance. The Benjamini-Hochberg method has been used to correct the p-value for multiple testing. Additionally, the top N features by L2FC for each cluster were kept after features in this table were filtered by (Mean UMI counts &gt; 1.0). Grayed-out features have an adjusted p-value &gt;= 0.10 or an L2FC &lt; 0. N (ranges from 1 to 50) is the number of top features displayed per cluster, which is set to limit the amount of table entries displayed to 10,000. N=%10,000/K^2 where K is the number of clusters. Click on a column to sort by that value, or search a gene of interest.</p>
<p></p><div class="alert alert-info hints-alert"><div class="hints-icon"><i class="fa fa-info-circle"></i></div><div class="hints-container"><p>Interpretation for exceptional cases related to differential expression analysis can be found under <a href="html-report.html#differential-expression-analysis-1">Square Bin</a> part.</p>
</div></div><p></p>
<h3 id="image-information"><a name="image-information" class="anchor-navigation-ex-anchor" href="#image-information"><i class="fa fa-link" aria-hidden="true"></i></a>Image information</h3>
<p>Basic information about the microscopic staining image, usually involving microscope settings.</p>
<h3 id="qc"><a name="qc" class="anchor-navigation-ex-anchor" href="#qc"><i class="fa fa-link" aria-hidden="true"></i></a>QC</h3>
<table><thead><tr><th width="232">Metric</th><th>Description</th></tr></thead><tbody><tr><td>Image QC version</td><td>The version of image QC module.</td></tr><tr><td>QC Pass</td><td>Whether the image(s) passed image QC quality check.</td></tr><tr><td>Trackline Score</td><td>Reference score for trackline detection.</td></tr></tbody></table>

<h3 id="stitching"><a name="stitching" class="anchor-navigation-ex-anchor" href="#stitching"><i class="fa fa-link" aria-hidden="true"></i></a>Stitching</h3>
<table><thead><tr><th width="251">Metric</th><th>Description</th></tr></thead><tbody><tr><td>Template Source Row No.</td><td>The row number of the template FOV used for predicting the entire template.</td></tr><tr><td>Template Source Column No.</td><td>The column number of the template FOV used for predicting the entire template.</td></tr><tr><td>Global Height</td><td>Height of the stitched image.</td></tr><tr><td>Global Width</td><td>Width of the stitched image.</td></tr></tbody></table>

<h3 id="registration"><a name="registration" class="anchor-navigation-ex-anchor" href="#registration"><i class="fa fa-link" aria-hidden="true"></i></a>Registration</h3>
<table><thead><tr><th width="256">Metric</th><th>Description</th></tr></thead><tbody><tr><td>ScaleX</td><td>The lateral scaling between image and template.</td></tr><tr><td>ScaleY</td><td>The longitudinal scaling between image and template.</td></tr><tr><td>Rotation</td><td>The rotation angle of the image relative to the template.</td></tr><tr><td>Flip</td><td>Whether the image is flipped horizontally.</td></tr><tr><td>Image X Offset</td><td>Offset between image and matrix in x direction.</td></tr><tr><td>Image Y Offset</td><td>Offset between image and matrix in y direction</td></tr><tr><td>Counter Clockwise Rotation</td><td>Counter clockwise rotation angle.</td></tr><tr><td>Manual ScaleX</td><td>The lateral scaling based on image center (manual-registration).</td></tr><tr><td>Manual ScaleY</td><td>The longitudinal scaling based on image center (manual-registration).</td></tr><tr><td>Manual Rotation</td><td>The rotation angle based on image center (manual-registration).</td></tr><tr><td>Matrix X Start</td><td>Gene expression matrix offset in x direction by DNB numbers.</td></tr><tr><td>Matrix Y Start</td><td>Gene expression matrix offset in y direction by DNB numbers.</td></tr><tr><td>Matrix Height</td><td>Gene expression matrix height.</td></tr><tr><td>Matrix Width</td><td>Gene expression matrix width.</td></tr></tbody></table>

<h2 id="microorganism"><a name="microorganism" class="anchor-navigation-ex-anchor" href="#microorganism"><i class="fa fa-link" aria-hidden="true"></i></a><strong>Microorganism</strong></h2>
<p></p><div class="alert alert-undefined hints-alert"><div class="hints-icon"><i class="undefined"></i></div><div class="hints-container"><p>Here is an another FFPE tissue sample of mouse lung which is especially for microorganism analysis.</p>
</div></div><p></p>
<figure><a href="../../img/assets/microorganism-expression.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/microorganism-expression.PNG" alt></a><figcaption><p>Microorganism heatmap under tissue region and four key metrics</p></figcaption></figure>

<p>The distribution plot of microorganism spatial expression, on the left, shows MID count at each bin20.</p>
<h3 id="denoising"><a name="denoising" class="anchor-navigation-ex-anchor" href="#denoising"><i class="fa fa-link" aria-hidden="true"></i></a>Denoising</h3>
<table><thead><tr><th width="244">Metric</th><th>Description</th></tr></thead><tbody><tr><td>Total Reads</td><td>Total number of input reads. (Ummapped Reads from transcriptome alignment)</td></tr><tr><td>Non-Host Source Reads</td><td>Number of reads that can not be aligned to the host genome.</td></tr><tr><td>Host Source Reads</td><td>Number of reads that can be aligned to the host genome during denoising.</td></tr></tbody></table>

<h3 id="taxonomic-classification"><a name="taxonomic-classification" class="anchor-navigation-ex-anchor" href="#taxonomic-classification"><i class="fa fa-link" aria-hidden="true"></i></a>Taxonomic Classification</h3>
<figure><a href="../../img/assets/microorganism-taxonnomy.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/microorganism-taxonnomy.PNG" alt></a><figcaption><p>Mapping result of Bowtie2 and Kraken2</p></figcaption></figure>

<table><thead><tr><th width="264">Metric</th><th>Description</th></tr></thead><tbody><tr><td>Non-Host Source Reads</td><td>Number of reads that can not be aligned to the host genome.</td></tr><tr><td>Bacteria, Fungi and Viruses MIDs</td><td>Number of unique mRNA molecular assigned to bacteria, fungi or viruses.</td></tr><tr><td>Bacteria, Fungi and Viruses Duplication</td><td>Number of assigned reads that have been corrected due to duplicated MID.</td></tr><tr><td>Other Microbes or Host-Suspicious</td><td>Number of reads assigned to other microbes (exclude bacteria, fungi and viruses) or host.</td></tr><tr><td>Unclassified Reads</td><td>Number of unclassified reads.</td></tr></tbody></table>

<h3 id="microbes-proportion-phylum"><a name="microbes-proportion-phylum" class="anchor-navigation-ex-anchor" href="#microbes-proportion-phylum"><i class="fa fa-link" aria-hidden="true"></i></a>Microbes Proportion (Phylum)</h3>
<figure><a href="../../img/assets/microorganism-proportion.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/microorganism-proportion.PNG" alt></a><figcaption><p>Microbes proportion at phylum level</p></figcaption></figure>

<p>The main proportion of microbes at the phylum level.</p>
<p><em>*the same for other classifications</em></p>
<h2 id="summary-protein"><a name="summary-protein" class="anchor-navigation-ex-anchor" href="#summary-protein"><i class="fa fa-link" aria-hidden="true"></i></a>Summary-Protein</h2>
<figure><a href="../../img/assets/summary-protein-expression%20(1).PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/summary-protein-expression (1).PNG" alt></a><figcaption></figcaption></figure>

<figure><a href="../../img/assets/summary-protein-image%20(2).PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/summary-protein-image (2).PNG" alt></a><figcaption></figcaption></figure>

<p>The spatial protein expression distribution plot, containing all bins and bins under tissue, on the left, shows MID count at each bin20.</p>
<p><code>Total Reads</code> is the total sequencing reads of input sequencing ADT FASTQs. <code>Valid CID reads</code> represents the number of reads with CIDs matching the mask file, with MIDs passing QC. <code>Valid PID reads</code> represents the number of reads that are mapped to the PID sequence in the protein panel. <code>Unique PID reads</code> represents the total number of unique protein reads (PID reads whose MIDs are different).</p>
<h3 id="key-metrics_1"><a name="key-metrics_1" class="anchor-navigation-ex-anchor" href="#key-metrics_1"><i class="fa fa-link" aria-hidden="true"></i></a>Key metrics</h3>
<figure><a href="../../img/assets/summary-protein-key-metrics.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/summary-protein-key-metrics.PNG" alt></a><figcaption></figcaption></figure>

<p>Key metrics of the data are listed:</p>
<table><thead><tr><th width="234">Metrics</th><th>Description</th></tr></thead><tbody><tr><td>Total Reads</td><td>Total number of sequenced reads</td></tr><tr><td>Valid CID Reads</td><td>Number of reads with CIDs matching the mask file and with MIDs passing QC</td></tr><tr><td>Invalid CID Reads</td><td>Number of reads with CIDs that cannot be matched with the mask file</td></tr><tr><td>Valid PID Reads</td><td>Valid CID reads that mapped to the protein sequence in the protein sequence database (protein panel)</td></tr><tr><td>Invalid PID Reads</td><td>Valid CID reads that can not be mapped to the protein sequence in the protein sequence database (protein panel)</td></tr><tr><td>Unique PID Reads</td><td>Total number of unique protein reads (PID reads whose MIDs are different)</td></tr><tr><td>Sequencing Saturation</td><td>Number of PID reads with duplicated MID</td></tr></tbody></table>

<h3 id="sequencing-saturation_1"><a name="sequencing-saturation_1" class="anchor-navigation-ex-anchor" href="#sequencing-saturation_1"><i class="fa fa-link" aria-hidden="true"></i></a>Sequencing saturation</h3>
<figure><a href="../../img/assets/summary-protein-saturation%20(1).PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/summary-protein-saturation (1).PNG" alt></a><figcaption></figcaption></figure>

<p>The saturation analysis in the HTML report can assess the overall quality of the sequencing data. In order to improve calculation efficiency, small samples are randomly selected from successfully annotated reads in the bin20 or bin50 dimension. Therefore, the results of multiple runs of the same data may vary slightly. The formulas may not be identical, but the general shape of the curve is consistent.</p>
<ul>
<li>Figure 1: Curves fitted based on Unique Reads data from randomly sampled samples.</li>
<li><strong>Figure 2</strong>: Statistics of Unique Reads (reads with unique CID, PID and MID) in the sampled samples, saturation value = 1-(Unique Reads)/(Valid PID Reads), as the sampling volume increases, the fitting curve becomes near-flat, indicating that the data tends to be saturated. Whether to add additional tests depends on the overall project design and sample conditions. For example, it is recommended that additional tests be performed on precious samples. The threshold value of 0.8 in the report serves as a reminder for recommended guidance.</li>
</ul>
<h3 id="protein-correlations"><a name="protein-correlations" class="anchor-navigation-ex-anchor" href="#protein-correlations"><i class="fa fa-link" aria-hidden="true"></i></a>Protein correlations</h3>
<p>Spearman correlation (in bin20 or bin50) between raw antibody counts under tissue, except isotype. Antibodies are clustered based on Spearman correlation coefficient.</p>
<figure><a href="../../img/assets/summary-protein-correlation%20(1).PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/summary-protein-correlation (1).PNG" alt></a><figcaption></figcaption></figure>

<h3 id="gene--protein-correlations"><a name="gene--protein-correlations" class="anchor-navigation-ex-anchor" href="#gene--protein-correlations"><i class="fa fa-link" aria-hidden="true"></i></a>Gene : protein correlations</h3>
<p>Spearman correlation (in bin20 or bin50) between raw gene counts and raw antibody counts under tissue, where antibody has at least one marker gene in the protein panel.</p>
<figure><a href="../../img/assets/summary-protein-gene-correlation.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/summary-protein-gene-correlation.PNG" alt></a><figcaption></figcaption></figure>

<h3 id="histogram-of-portein-counts"><a name="histogram-of-portein-counts" class="anchor-navigation-ex-anchor" href="#histogram-of-portein-counts"><i class="fa fa-link" aria-hidden="true"></i></a>Histogram of portein counts</h3>
<p>Distribution of spot numbers vs log-scaled MID count (in bin20 or bin50).</p>
<figure><a href="../../img/assets/summary-protein-histogram%20(1).PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/summary-protein-histogram (1).PNG" alt></a><figcaption></figcaption></figure>

<h3 id="information_1"><a name="information_1" class="anchor-navigation-ex-anchor" href="#information_1"><i class="fa fa-link" aria-hidden="true"></i></a>Information</h3>
<p>This item displays the basic information of the input FASTQs, </p>
<p><code>Species</code> is from the <code>--organism</code> parameter used in <code>SAW count</code>, usually referring to the species.</p>
<p><code>Tissue</code> is from the <code>--tissue</code> parameter used in <code>SAW count</code>.</p>
<p><code>Reference</code> means the reference genome used in <code>SAW count</code>, as the same as <code>Organism</code>.</p>
<p><code>FASTQ</code> records  FASTQ files in <code>SAW count</code>, including file prefixes of all input sequencing ADT FASTQs.</p>
<h2 id="square-bin-protein"><a name="square-bin-protein" class="anchor-navigation-ex-anchor" href="#square-bin-protein"><i class="fa fa-link" aria-hidden="true"></i></a>Square Bin-Protein</h2>
<p>This page contains results of statistics, plots, clustering, UMAP, and differential expression analysis, at bin dimension. Results come from the analysis based on <code>&lt;SN&gt;.protein.tissue.gef</code> file.</p>
<h3 id="statistics_2"><a name="statistics_2" class="anchor-navigation-ex-anchor" href="#statistics_2"><i class="fa fa-link" aria-hidden="true"></i></a>Statistics</h3>
<figure><a href="../../img/assets/squarebin-protein-table.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/squarebin-protein-table.PNG" alt></a><figcaption></figcaption></figure>

<table><thead><tr><th width="247">Item</th><th>Description</th></tr></thead><tbody><tr><td>Bin Size</td><td><p>The size of Bin which is the unit of aggregated DNBs in a squared region.</p><p>i.e. Bin 50 = 50 * 50 DNBs</p></td></tr><tr><td>Mean Reads (per bin)</td><td>Mean number of sequenced reads divided by the number of bins under tissue coverage.</td></tr><tr><td>Median Reads (per bin)</td><td>Median number of sequenced reads divided by the number of bins under tissue coverage (pick the middle value after sorting).</td></tr><tr><td>Mean MID (per bin)</td><td>Mean number of MIDs divided by the number of bins under tissue coverage.</td></tr><tr><td>Median MID (per bin)</td><td>Median number of MIDs divided by the number of bins under tissue coverage</td></tr></tbody></table>

<h3 id="plots_2"><a name="plots_2" class="anchor-navigation-ex-anchor" href="#plots_2"><i class="fa fa-link" aria-hidden="true"></i></a>Plots</h3>
<figure><a href="../../img/assets/squarebin-protein-violin.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/squarebin-protein-violin.PNG" alt></a><figcaption></figcaption></figure>

<p>Violin plots show the distribution of deduplicated MID count in each bin.</p>
<h3 id="clustering--umap_2"><a name="clustering--umap_2" class="anchor-navigation-ex-anchor" href="#clustering--umap_2"><i class="fa fa-link" aria-hidden="true"></i></a>Clustering &amp; UMAP</h3>
<figure><a href="../../img/assets/bin20-protein-cluster.png" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/bin20-protein-cluster.png" alt></a><figcaption></figcaption></figure>

<p>Clustering is performed based on <code>SN.protein.tissue.gef</code> using the Leiden algorithm. UMAP projections are performed based on <code>SN.protein.tissue.gef</code> and colored by automated clustering. The same color is assigned to spots that are within a shorter distance and with similar gene expression profiles.</p>
<h2 id="cell-bin-protein"><a name="cell-bin-protein" class="anchor-navigation-ex-anchor" href="#cell-bin-protein"><i class="fa fa-link" aria-hidden="true"></i></a>Cell Bin-Protein</h2>
<p>This page contains results of statistics, plots, clustering, UMAP, and differential expression analysis, at cellbin dimension. Cell border expanding is automatically performed during <code>SAW count</code> and <code>SAW realign</code>, which means the contents of &quot;Cell Bin&quot; tab are based on <code>SN.protein.adjusted.cellbin.gef</code>. </p>
<p></p><div class="alert alert-undefined hints-alert"><div class="hints-icon"><i class="undefined"></i></div><div class="hints-container"><p>When it comes to <code>--adjusted-distance=0</code> in <code>SAW realign</code>, all contents of this tab are based on <code>SN.protein.cellbin.gef</code>. </p>
</div></div><p></p>
<h3 id="statistics_3"><a name="statistics_3" class="anchor-navigation-ex-anchor" href="#statistics_3"><i class="fa fa-link" aria-hidden="true"></i></a>Statistics</h3>
<figure><a href="../../img/assets/cellbin-protein-table%20(1).PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/cellbin-protein-table (1).PNG" alt></a><figcaption></figcaption></figure>

<p>The above table records the statistics of cellbin:</p>
<table><thead><tr><th width="247">Item</th><th>Description</th></tr></thead><tbody><tr><td>Cell Count</td><td>Number of cells.</td></tr><tr><td>Mean Cell Area</td><td>Mean cell area, in pixes.</td></tr><tr><td>Median Cell Area</td><td>Median cell area, in pixes.</td></tr><tr><td>Mean MID</td><td>Mean MID count per cell.</td></tr><tr><td>Median MID</td><td>Median MID count per cell.</td></tr></tbody></table>

<h3 id="plots_3"><a name="plots_3" class="anchor-navigation-ex-anchor" href="#plots_3"><i class="fa fa-link" aria-hidden="true"></i></a>Plots</h3>
<figure><a href="../../img/assets/cellbin-protein-violin.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/cellbin-protein-violin.PNG" alt></a><figcaption></figcaption></figure>

<p>Violin plots show the distribution of deduplicated MID count and cell area in the cellbin.</p>
<h3 id="clustering--umap_3"><a name="clustering--umap_3" class="anchor-navigation-ex-anchor" href="#clustering--umap_3"><i class="fa fa-link" aria-hidden="true"></i></a>Clustering &amp; UMAP</h3>
<figure><a href="../../img/assets/cellbin-protein-cluster.PNG" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/cellbin-protein-cluster.PNG" alt></a><figcaption><p>Leiden clustering and UMAP projection</p></figcaption></figure>

<p>Clustering is performed based on <code>SN.protein.adjusted.cellbin.gef</code> or <code>SN.protein.cellbin.gef</code>, using the Leiden algorithm. UMAP projections are performed based on <code>SN.protein.adjusted.cellbin.gef</code> or <code>SN.protein.cellbin.gef</code>, and colored by automated clustering. The same color is assigned to spots that are within a shorter distance and with similar gene expression profiles.</p>
<h2 id="analysis-no-longer-available-since-812"><a name="analysis-no-longer-available-since-812" class="anchor-navigation-ex-anchor" href="#analysis-no-longer-available-since-812"><i class="fa fa-link" aria-hidden="true"></i></a>Analysis (no longer available since 8.1.2)</h2>
<p>This page contains results of clustering from gene-protein joint analysis, marker selection, and correlation between genes and proteins. </p>
<h3 id="multiomics-clustering--umap"><a name="multiomics-clustering--umap" class="anchor-navigation-ex-anchor" href="#multiomics-clustering--umap"><i class="fa fa-link" aria-hidden="true"></i></a>Multiomics clustering &amp; UMAP</h3>
<p>Clustering and UMAP projections are performed based on the latent space generated by totalVI.</p>
<h3 id="top-markers-by-cluster"><a name="top-markers-by-cluster" class="anchor-navigation-ex-anchor" href="#top-markers-by-cluster"><i class="fa fa-link" aria-hidden="true"></i></a>Top markers by cluster</h3>
<p>Heatmaps of top &lt;=3 gene and protein markers per Leiden cluster from gene-protein jointly analysis. These features are filtered after one-vs-all differential expression analysis, following these rules:</p>
<ul>
<li>For gene, Bayes factor &gt; 1 and  expression proportion greater than 10% in the cluster;</li>
<li>For protein, Bayes factor &gt; 0.7.</li>
</ul>
<h2 id="alerts"><a name="alerts" class="anchor-navigation-ex-anchor" href="#alerts"><i class="fa fa-link" aria-hidden="true"></i></a><strong>Alerts</strong></h2>
<p>Thresholds are set for several important statistical indicators. If the analysis results are abnormal, an alert message will be displayed at the top of the HTML report.</p>
<p></p><div class="alert alert-undefined hints-alert"><div class="hints-icon"><i class="undefined"></i></div><div class="hints-container"><p>Here is an abnormal exmple data just for display.</p>
</div></div><p></p>
<figure><a href="../../img/assets/image%20(80).png" data-lightbox="54c83b9a-d310-4a2e-95be-b7f678a437e4" data-title><img src="../../img/assets/image (80).png" alt></a><figcaption><p>Alert information</p></figcaption></figure>


<footer class="page-footer"><span class="copyright">&#xA9; 2025 STOmics Tech. All rights reserved.</span><span class="footer-modification">Modified: 
2025-04-09 15:29:46
</span></footer><script>document.addEventListener("DOMContentLoaded", function() {lightbox.option({"resizeDuration":200,"wrapAround":true});})</script>
                                
                                </section>
                            
    </div>
    <div class="search-results">
        <div class="has-results">
            
            <h1 class="search-results-title"><span class="search-results-count"></span> results matching &quot;<span class="search-query"></span>&quot;</h1>
            <ul class="search-results-list"></ul>
            
        </div>
        <div class="no-results">
            
            <h1 class="search-results-title">No results matching &quot;<span class="search-query"></span>&quot;</h1>
            
        </div>
    </div>
</div>

                        </div>
                    </div>
                
            </div>

            
                
                <a href="realign-outputs.html" class="navigation navigation-prev " aria-label="Previous page: realign &#x8F93;&#x51FA;&#x7ED3;&#x679C;">
                    <i class="fa fa-angle-left"></i>
                </a>
                
                
                <a href="bam.html" class="navigation navigation-next " aria-label="Next page: BAM">
                    <i class="fa fa-angle-right"></i>
                </a>
                
            
        
    </div>

    <script>
        var gitbook = gitbook || [];
        gitbook.push(function() {
            gitbook.page.hasChanged({"page":{"title":"HTML report","level":"1.4.3.3","depth":3,"next":{"title":"BAM","level":"1.4.3.4","depth":3,"path":"analysis/outputs/bam.md","ref":"analysis/outputs/bam.md","articles":[]},"previous":{"title":"realign ","level":"1.4.3.2","depth":3,"path":"analysis/outputs/realign-outputs.md","ref":"analysis/outputs/realign-outputs.md","articles":[]},"dir":"ltr"},"config":{"plugins":["expandable-chapters-small","anchor-navigation-ex","logo","-insert-logo","tbfed-pagefooter","hide-element","-code-optimize","katex","hints","tabs","lightbox","-lunr","-search","search-plus","add-index-html","anchor-to-full-links"],"root":".","styles":{},"pluginsConfig":{"tbfed-pagefooter":{"copyright":" 2025 STOmics Tech.","modify_label":"Modified: ","modify_format":"YYYY-MM-DD HH:mm:ss"},"logo":{"logo":"data:image/png;base64,AAAAGGZ0eXBhdmlmAAAAAG1pZjFtaWFmAAAA0m1ldGEAAAAAAAAAIWhkbHIAAAAAAAAAAHBpY3QAAAAAAAAAAAAAAAAAAAAADnBpdG0AAAAAAAEAAAAeaWxvYwAAAABEAAABAAEAAAABAAAA8gAABgAAAAAjaWluZgAAAAAAAQAAABVpbmZlAgAAAAABAABhdjAxAAAAAFZpcHJwAAAAOGlwY28AAAAUaXNwZQAAAAAAAAAgAAAALQAAAAxhdjFDgT9AAAAAABBwaXhpAAAAAAMKCgoAAAAWaXBtYQAAAAAAAAABAAEDAYIDAAAGCG1kYXQSAAoJP9F/YXgIaDaAMvALZAQYAAQAAgIAAAAAAAAAEAAAAAVACJZH5jEfcnHq5GfVOqfqAcroK/CW/n75YIMe8ZfAjZNFE/6UrpMmpDP/er+E0q/ceflnZGp4dyYbk4GCBlbhdKT3x+D/AIaisuTFd8nYmo+ez1m+B7mjO9pucuLd4J09fZofEcRZ1N8yXb1uByDJRlxGRNCsg51xqZYiGzKHoIgR4ChwET50S4FSgHUScr3K3YmkXZjZK1Sle8EZjq6uDLYWGw4NvJ/pvNMkd/WmxjuUTT7pfVtNIGjG28GzYzHlf3vaTaGjRUC91DO9vrJT51k9bZZg8h2LiVbz7yNTOZcjlBLZuokz15s00uLVyWobV2CGySmbuxBbAsKxItjkVbSHo6ojiJbY5bPJvksg0SaezQgA1EgcN96PcNp/fjOFRBtevV/O+rjq23jHY8C9n/YSCdXnGsWJNHdxleLb40bjcNXoXncQADGMuX4XOfKPzQ3L6GO6nlgQBXZyDdvAAbF778o6nSAV86cHKEWnD3tckPflzq+6jvx4oCjObYt0X1Ro63jOf55vCFAlxB8FJNtvuok/MW0bSKvp+0LTGESD/2lmWnH+IvOX5hbFmtTw5mgGW0eKFHyZXqg00hWU4wfzKPxpRwvxAUZMaiFlEudhK9HjhFYdgz/1ShJlEUXzbR+U6nabVjctZv4ZQeECNYZ3p4pAGxtsSlxZBYQ4ARr8zqcDQjsYk+rr1KdIQ14PnyCIe0imW/rJ/at5+YUvZ6Z4FApMELY7udGJJ5HuVaPCr22ij/h9mvhjgSMMfCesXRgyxDAWg9Aa+lx/LxDWkXLxHot/ByjPyng7o28F0TwpKuvCU8x8MxSuy9o89zSiGVR2nsz7EEWJvFEeZKRMn9j30VIwqcF4D/tWMO3puIoih0v/CvKusdf7v0qTb58ZaIGvRrTJSLiosclEEtiORGa0wihqVOcL7O3GL2rIRP/Pcv1BSDYXhybnLZ+HbhjuBO2ulzWv3j+0GWtGdtmij8t9ivPRKOC7RYXLQ99mYd621NSxNWbNJaP1XYbLFkbATt+nBDr+ypqyy+ARzVlE9ywft4Zj683T2knXPPhbo0X9I70mu1LIAsFCr9IwunPhTV//P4W4w8S1Ks9IB7NeBdiWCCtW1KWtIuFaIPvx+VbZkxyad+Nqh4iFTKBYIfGffIbiEQiBaJaxB+uIC8/TylgtlADj00ipv0HgSHv3v2rna/hOViWRbA9kars4YRbv8GF4i4delTz8Myp9DsZQv/owkRFJkk5KHrv25DT0CrW2FxlBIuJQuSHak+OCRtl+RnJwNd+0wDZ09WRFCkbt6qoCbVumBWzBM/0afWjK8QukftmXeX93+lbWGWrOova/TResZhzunNubiTDOH6WgLh/g8v3YUg1S5V2GzESjFUtpQBZFiQVn6a+beAxr3kQINzRSASF1nZZjdiEmN7rXfuhoqkBJV6uARAJBDfPNARGsjTgivGjqjHdjeZHNowz9apljGORFaJJBbE0zTjcd7NUmAiSd9dIXG8cAmiBmmU1fUbaupjLvmwgxdI9ZxfpP3h7WygcJklKiA+Unx+LJT20S7nW1nO/HiqEKOAnhKKEvpWo8NWVVJ1l1HXanmZRnuA+DkpOr4ahp63B1LxmbWnEP8IzsZtbQWG1+yQUfwN4CYnnPZB34ljqnPbGb+ReVnQ/B552w6HkZFlnSYxvuVvAg6LnzslXN9UZZreNj6xTh9AAsznkSmpb7//5b4QpSnH9N948K/lP3QND+g6Co5XKGmAfUS+NnBs4/AzZQg+3n/VjAIHA5YIMMduPmXmlwzDiQOZC2pSYfpMUqIScTuITS35U5C8lq86m1jGD1ZSZRse4fJdFbPKs+TldyLZyqXub516/e1dRG47gRwSUoWfcjoG486GAZTabIiCOxD9Pq13HejOuZG+j+/0zALlXdSctzuOMx1XJMaYHpJ06kZmkCGN+JXeuDjk32LVGFUn/WY7XCw1tvJQBykA6dQTtjhbrhALglaNHrFiw=","title":"SAW  8.1 ","titleStyle":"font-weight:600"},"hints":{"danger":"fa fa-exclamation-circle","info":"fa fa-info-circle","tip":"fa fa-mortar-board","working":"fa fa-wrench"},"hide-element":{"elements":[".gitbook-link"]},"katex":{},"add-index-html":{"myConfigKey":"it's the default value","elements":["a"]},"fontsettings":{"theme":"white","family":"sans","size":2},"highlight":{},"anchor-navigation-ex":{"associatedWithSummary":true,"float":{"floatIcon":"fa fa-navicon","level1Icon":"fa fa-hand-o-right","level2Icon":"fa fa-hand-o-right","level3Icon":"fa fa-hand-o-right","showLevelIcon":false},"mode":"float","multipleH1":true,"pageTop":{"level1Icon":"fa fa-hand-o-right","level2Icon":"fa fa-hand-o-right","level3Icon":"fa fa-hand-o-right","showLevelIcon":false},"printLog":true,"showGoTop":true,"showLevel":false},"anchor-to-full-links":{"myConfigKey":"it's the default value","elements":["a"]},"lightbox":{"jquery":true,"includeJQuery":true,"sameUuid":true,"options":{"resizeDuration":200,"wrapAround":true}},"expandable-chapters-small":{},"sharing":{"facebook":false,"twitter":false,"google":false,"weibo":false,"instapaper":false,"vk":false,"all":[]},"theme-default":{"styles":{"website":"styles/website.css","pdf":"styles/pdf.css","epub":"styles/epub.css","mobi":"styles/mobi.css","ebook":"styles/ebook.css","print":"styles/print.css"},"showLevel":false},"code-optimize":{"copyButtons":true,"showLines":true},"tabs":{},"search-plus":{}},"theme":"default","author":"","pdf":{"pageBreaksBefore":"/","headerTemplate":null,"paperSize":"a4","margin":{"right":62,"left":62,"top":36,"bottom":36},"fontSize":12,"fontFamily":"Arial","footerTemplate":null,"chapterMark":"pagebreak","pageNumbers":false},"structure":{"langs":"LANGS.md","readme":"README.md","glossary":"GLOSSARY.md","summary":"SUMMARY.md"},"isbn":"","variables":{},"title":"SAW User Manual","language":"zh-hans","links":{"gitbook":false,"sharing":{"google":false,"facebook":false,"twitter":false,"all":false}},"gitbook":"3.2.3","description":null,"extension":null},"file":{"path":"analysis/outputs/html-report.md","mtime":"2025-04-09T07:29:46.835Z","type":"markdown"},"gitbook":{"version":"3.2.3","time":"2025-04-09T09:29:48.718Z"},"basePath":"../..","book":{"language":""}});
        });
    </script>
</div>

        
    <script src="../../gitbook/gitbook.js"></script>
    <script src="../../gitbook/theme.js"></script>
    
        
        <script src="../../gitbook/gitbook-plugin-expandable-chapters-small/expandable-chapters-small.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-hide-element/plugin.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-tabs/bootstrap.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-lightbox/js/lightbox.min.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-search-plus/jquery.mark.min.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-search-plus/search.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-add-index-html/plugin.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-anchor-to-full-links/plugin.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-sharing/buttons.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-fontsettings/fontsettings.js"></script>
        
    

    </body>
</html>

