
<!DOCTYPE HTML>
<html lang="zh-hans">
    <head>
        <meta charset="UTF-8">
        <meta content="text/html; charset=utf-8" http-equiv="Content-Type">
        <title>Read processing algorithms &#xB7; SAW User Manual</title>
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="description" content>
        <meta name="generator" content="GitBook 3.2.3">
        
        
        
    
    <link rel="stylesheet" href="../gitbook/style.css">

    
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-expandable-chapters-small/expandable-chapters-small.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-anchor-navigation-ex/style/plugin.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-tbfed-pagefooter/footer.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-katex/katex.min.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-hints/plugin-hints.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-tabs/bootstrap.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-lightbox/css/lightbox.min.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-search-plus/search.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-highlight/website.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-fontsettings/website.css">
                
            
        

    

    

        
    
    
    <meta name="HandheldFriendly" content="true">
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <link rel="apple-touch-icon-precomposed" sizes="152x152" href="../gitbook/images/apple-touch-icon-precomposed-152.png">
    <link rel="shortcut icon" href="../gitbook/images/favicon.ico" type="image/x-icon">

    
    <link rel="next" href="image-processing-algorithms.html">
    
    
    <link rel="prev" href="./">
    

    </head>
    <body>
        
<div class="book">
    <div class="book-summary"><div class="book-logo-root"><div class="book-logo" style="text-align: center; padding: 20px;"><img src="data:image/png;base64,AAAAGGZ0eXBhdmlmAAAAAG1pZjFtaWFmAAAA0m1ldGEAAAAAAAAAIWhkbHIAAAAAAAAAAHBpY3QAAAAAAAAAAAAAAAAAAAAADnBpdG0AAAAAAAEAAAAeaWxvYwAAAABEAAABAAEAAAABAAAA8gAABgAAAAAjaWluZgAAAAAAAQAAABVpbmZlAgAAAAABAABhdjAxAAAAAFZpcHJwAAAAOGlwY28AAAAUaXNwZQAAAAAAAAAgAAAALQAAAAxhdjFDgT9AAAAAABBwaXhpAAAAAAMKCgoAAAAWaXBtYQAAAAAAAAABAAEDAYIDAAAGCG1kYXQSAAoJP9F/YXgIaDaAMvALZAQYAAQAAgIAAAAAAAAAEAAAAAVACJZH5jEfcnHq5GfVOqfqAcroK/CW/n75YIMe8ZfAjZNFE/6UrpMmpDP/er+E0q/ceflnZGp4dyYbk4GCBlbhdKT3x+D/AIaisuTFd8nYmo+ez1m+B7mjO9pucuLd4J09fZofEcRZ1N8yXb1uByDJRlxGRNCsg51xqZYiGzKHoIgR4ChwET50S4FSgHUScr3K3YmkXZjZK1Sle8EZjq6uDLYWGw4NvJ/pvNMkd/WmxjuUTT7pfVtNIGjG28GzYzHlf3vaTaGjRUC91DO9vrJT51k9bZZg8h2LiVbz7yNTOZcjlBLZuokz15s00uLVyWobV2CGySmbuxBbAsKxItjkVbSHo6ojiJbY5bPJvksg0SaezQgA1EgcN96PcNp/fjOFRBtevV/O+rjq23jHY8C9n/YSCdXnGsWJNHdxleLb40bjcNXoXncQADGMuX4XOfKPzQ3L6GO6nlgQBXZyDdvAAbF778o6nSAV86cHKEWnD3tckPflzq+6jvx4oCjObYt0X1Ro63jOf55vCFAlxB8FJNtvuok/MW0bSKvp+0LTGESD/2lmWnH+IvOX5hbFmtTw5mgGW0eKFHyZXqg00hWU4wfzKPxpRwvxAUZMaiFlEudhK9HjhFYdgz/1ShJlEUXzbR+U6nabVjctZv4ZQeECNYZ3p4pAGxtsSlxZBYQ4ARr8zqcDQjsYk+rr1KdIQ14PnyCIe0imW/rJ/at5+YUvZ6Z4FApMELY7udGJJ5HuVaPCr22ij/h9mvhjgSMMfCesXRgyxDAWg9Aa+lx/LxDWkXLxHot/ByjPyng7o28F0TwpKuvCU8x8MxSuy9o89zSiGVR2nsz7EEWJvFEeZKRMn9j30VIwqcF4D/tWMO3puIoih0v/CvKusdf7v0qTb58ZaIGvRrTJSLiosclEEtiORGa0wihqVOcL7O3GL2rIRP/Pcv1BSDYXhybnLZ+HbhjuBO2ulzWv3j+0GWtGdtmij8t9ivPRKOC7RYXLQ99mYd621NSxNWbNJaP1XYbLFkbATt+nBDr+ypqyy+ARzVlE9ywft4Zj683T2knXPPhbo0X9I70mu1LIAsFCr9IwunPhTV//P4W4w8S1Ks9IB7NeBdiWCCtW1KWtIuFaIPvx+VbZkxyad+Nqh4iFTKBYIfGffIbiEQiBaJaxB+uIC8/TylgtlADj00ipv0HgSHv3v2rna/hOViWRbA9kars4YRbv8GF4i4delTz8Myp9DsZQv/owkRFJkk5KHrv25DT0CrW2FxlBIuJQuSHak+OCRtl+RnJwNd+0wDZ09WRFCkbt6qoCbVumBWzBM/0afWjK8QukftmXeX93+lbWGWrOova/TResZhzunNubiTDOH6WgLh/g8v3YUg1S5V2GzESjFUtpQBZFiQVn6a+beAxr3kQINzRSASF1nZZjdiEmN7rXfuhoqkBJV6uARAJBDfPNARGsjTgivGjqjHdjeZHNowz9apljGORFaJJBbE0zTjcd7NUmAiSd9dIXG8cAmiBmmU1fUbaupjLvmwgxdI9ZxfpP3h7WygcJklKiA+Unx+LJT20S7nW1nO/HiqEKOAnhKKEvpWo8NWVVJ1l1HXanmZRnuA+DkpOr4ahp63B1LxmbWnEP8IzsZtbQWG1+yQUfwN4CYnnPZB34ljqnPbGb+ReVnQ/B552w6HkZFlnSYxvuVvAg6LnzslXN9UZZreNj6xTh9AAsznkSmpb7//5b4QpSnH9N948K/lP3QND+g6Co5XKGmAfUS+NnBs4/AzZQg+3n/VjAIHA5YIMMduPmXmlwzDiQOZC2pSYfpMUqIScTuITS35U5C8lq86m1jGD1ZSZRse4fJdFbPKs+TldyLZyqXub516/e1dRG47gRwSUoWfcjoG486GAZTabIiCOxD9Pq13HejOuZG+j+/0zALlXdSctzuOMx1XJMaYHpJ06kZmkCGN+JXeuDjk32LVGFUn/WY7XCw1tvJQBykA6dQTtjhbrhALglaNHrFiw=" style><p class="book-title" style="font-weight:600">SAW &#x7528;&#x6237;&#x624B;&#x518C; 8.1 </p></div></div>
        
            
<div id="book-search-input" role="search">
    <input type="text" placeholder="&#x8F93;&#x5165;&#x5E76;&#x641C;&#x7D22;">
</div>

            
                <nav role="navigation">
                


<ul class="summary">
    
    

    

    
        
        
    
        <li class="chapter " data-level="1.1" data-path="../">
            
                <a href="../">
            
                    
                    &#x6982;&#x89C8;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.2" data-path="../getting-started/">
            
                <a href="../getting-started/">
            
                    
                    &#x5FEB;&#x901F;&#x5F00;&#x59CB;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.2.1" data-path="../getting-started/introduction.html">
            
                <a href="../getting-started/introduction.html">
            
                    
                    &#x8F6F;&#x4EF6;&#x4ECB;&#x7ECD;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.2.2" data-path="../getting-started/glossary.html">
            
                <a href="../getting-started/glossary.html">
            
                    
                    &#x8BCD;&#x6C47;&#x8868;
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.3" data-path="../download-center.html">
            
                <a href="../download-center.html">
            
                    
                    &#x4E0B;&#x8F7D;&#x4E2D;&#x5FC3;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4" data-path="../analysis/">
            
                <a href="../analysis/">
            
                    
                    &#x8FD0;&#x884C;&#x5206;&#x6790;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.1" data-path="../analysis/inputs/">
            
                <a href="../analysis/inputs/">
            
                    
                    &#x8F93;&#x5165;&#x6587;&#x4EF6;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.1.1" data-path="../analysis/inputs/chip-mask.html">
            
                <a href="../analysis/inputs/chip-mask.html">
            
                    
                    &#x82AF;&#x7247; mask
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.1.2" data-path="../analysis/inputs/fastqs.html">
            
                <a href="../analysis/inputs/fastqs.html">
            
                    
                    &#x6D4B;&#x5E8F; FASTQ
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.1.3" data-path="../analysis/inputs/images.html">
            
                <a href="../analysis/inputs/images.html">
            
                    
                    &#x663E;&#x5FAE;&#x955C;&#x56FE;&#x50CF;
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.4.2" data-path="../analysis/pipelines/">
            
                <a href="../analysis/pipelines/">
            
                    
                    &#x5206;&#x6790;&#x6D41;&#x7A0B;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.2.1" data-path="../analysis/pipelines/saw-can-shu-ming-ling.html">
            
                <a href="../analysis/pipelines/saw-can-shu-ming-ling.html">
            
                    
                    SAW &#x53C2;&#x6570;&#x547D;&#x4EE4;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.2.2" data-path="../analysis/pipelines/count/">
            
                <a href="../analysis/pipelines/count/">
            
                    
                    count
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.2.2.1" data-path="../analysis/pipelines/count/stereo-seq-t-ff.html">
            
                <a href="../analysis/pipelines/count/stereo-seq-t-ff.html">
            
                    
                    Stereo-seq T FF
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.2.2.2" data-path="../analysis/pipelines/count/stereo-seq-n-ffpe-v1.0.html">
            
                <a href="../analysis/pipelines/count/stereo-seq-n-ffpe-v1.0.html">
            
                    
                    Stereo-seq N FFPE
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.2.2.3" data-path="../analysis/pipelines/count/stereo-cite-t-ff.html">
            
                <a href="../analysis/pipelines/count/stereo-cite-t-ff.html">
            
                    
                    Stereo-CITE T FF
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.4.3" data-path="../analysis/outputs/">
            
                <a href="../analysis/outputs/">
            
                    
                    &#x8F93;&#x51FA;&#x7ED3;&#x679C;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.3.1" data-path="../analysis/outputs/count-outputs.html">
            
                <a href="../analysis/outputs/count-outputs.html">
            
                    
                    count &#x8F93;&#x51FA;&#x7ED3;&#x679C;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.2" data-path="../analysis/outputs/realign-outputs.html">
            
                <a href="../analysis/outputs/realign-outputs.html">
            
                    
                    realign &#x8F93;&#x51FA;&#x7ED3;&#x679C;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.3" data-path="../analysis/outputs/html-report.html">
            
                <a href="../analysis/outputs/html-report.html">
            
                    
                    HTML report
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.4" data-path="../analysis/outputs/bam.html">
            
                <a href="../analysis/outputs/bam.html">
            
                    
                    BAM
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.5" data-path="../analysis/outputs/matrices.html">
            
                <a href="../analysis/outputs/matrices.html">
            
                    
                    Matrices
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.6" data-path="../analysis/outputs/images.html">
            
                <a href="../analysis/outputs/images.html">
            
                    
                    Images
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.7" data-path="../analysis/outputs/analysis.html">
            
                <a href="../analysis/outputs/analysis.html">
            
                    
                    Analysis
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.5" data-path="../shi-yong-jiao-cheng/">
            
                <a href="../shi-yong-jiao-cheng/">
            
                    
                    &#x4F7F;&#x7528;&#x6559;&#x7A0B;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.5.1" data-path="../shi-yong-jiao-cheng/preparation-of-reference.html">
            
                <a href="../shi-yong-jiao-cheng/preparation-of-reference.html">
            
                    
                    &#x6BD4;&#x5BF9;&#x7D22;&#x5F15;&#x6784;&#x5EFA;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.2" data-path="../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/">
            
                <a href="../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/">
            
                    
                    &#x8FD0;&#x884C;&#x6838;&#x5FC3;&#x5206;&#x6790;&#x6D41;&#x7A0B;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.5.2.1" data-path="../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/stereo-seq-ff.html">
            
                <a href="../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/stereo-seq-ff.html">
            
                    
                    Stereo-seq FF &#x5206;&#x6790;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.2.2" data-path="../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/stereo-seq-ffpe.html">
            
                <a href="../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/stereo-seq-ffpe.html">
            
                    
                    Stereo-seq FFPE &#x5206;&#x6790;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.2.3" data-path="../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/stereocite-ff-fen-xi.html">
            
                <a href="../shi-yong-jiao-cheng/yun-xing-he-xin-fen-xi-liu-cheng/stereocite-ff-fen-xi.html">
            
                    
                    Stereo-CITE FF &#x5206;&#x6790;
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.5.3" data-path="../shi-yong-jiao-cheng/shou-dong-tu-xiang-chu-li.html">
            
                <a href="../shi-yong-jiao-cheng/shou-dong-tu-xiang-chu-li.html">
            
                    
                    &#x624B;&#x52A8;&#x56FE;&#x50CF;&#x5904;&#x7406;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.4" data-path="../shi-yong-jiao-cheng/secondary-analysis.html">
            
                <a href="../shi-yong-jiao-cheng/secondary-analysis.html">
            
                    
                    &#x6570;&#x636E;&#x518D;&#x5206;&#x6790;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.5" data-path="../shi-yong-jiao-cheng/shu-ju-ge-shi-zhuan-huan.html">
            
                <a href="../shi-yong-jiao-cheng/shu-ju-ge-shi-zhuan-huan.html">
            
                    
                    &#x6570;&#x636E;&#x683C;&#x5F0F;&#x8F6C;&#x6362;
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.6" data-path="./">
            
                <a href="./">
            
                    
                    &#x7B97;&#x6CD5;&#x539F;&#x7406;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter active" data-level="1.6.1" data-path="gene-expression-algorithms.html">
            
                <a href="gene-expression-algorithms.html">
            
                    
                    Read processing algorithms
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.6.2" data-path="image-processing-algorithms.html">
            
                <a href="image-processing-algorithms.html">
            
                    
                    Image processing algorithms
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.7" data-path="../gao-ji-she-zhi/">
            
                <a href="../gao-ji-she-zhi/">
            
                    
                    &#x9AD8;&#x7EA7;&#x8BBE;&#x7F6E;
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.7.1" data-path="../gao-ji-she-zhi/analysis-with-gpu.html">
            
                <a href="../gao-ji-she-zhi/analysis-with-gpu.html">
            
                    
                    Analysis with GPU
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.7.2" data-path="../gao-ji-she-zhi/expression-matrix-format.html">
            
                <a href="../gao-ji-she-zhi/expression-matrix-format.html">
            
                    
                    Expression matrix format
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.7.3" data-path="../gao-ji-she-zhi/image-related-record-format.html">
            
                <a href="../gao-ji-she-zhi/image-related-record-format.html">
            
                    
                    Image-related record format
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.8" data-path="../release-notes.html">
            
                <a href="../release-notes.html">
            
                    
                    &#x7248;&#x672C;&#x8BF4;&#x660E;
            
                </a>
            

            
        </li>
    

    
        
        <li class="header">&#x6280;&#x672F;&#x652F;&#x6301;</li>
        
        
    
        <li class="chapter " data-level="2.1" data-path="../support/shu-ju-ji.html">
            
                <a href="../support/shu-ju-ji.html">
            
                    
                    &#x6570;&#x636E;&#x96C6;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="2.2" data-path="../support/cuo-wu-ma.html">
            
                <a href="../support/cuo-wu-ma.html">
            
                    
                    &#x9519;&#x8BEF;&#x7801;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="2.3" data-path="../support/shi-pei-jian-rong-de-ruan-jian-he-shi-ji-he.html">
            
                <a href="../support/shi-pei-jian-rong-de-ruan-jian-he-shi-ji-he.html">
            
                    
                    &#x9002;&#x914D;&#x517C;&#x5BB9;&#x7684;&#x8F6F;&#x4EF6;&#x548C;&#x8BD5;&#x5242;&#x76D2;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="2.4">
            
                <a target="_blank" href="https://www.stomics.tech/resources/FAQ">
            
                    
                    &#x5E38;&#x89C1;&#x95EE;&#x9898;
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="2.5">
            
                <a target="_blank" href="https://www.stomics.tech/service/new-StereoMap.html">
            
                    
                    StereoMap
            
                </a>
            

            
        </li>
    

    

    <li class="divider"></li>

    <li>
        <a href="https://www.gitbook.com" target="blank" class="gitbook-link">
            &#x672C;&#x4E66;&#x4F7F;&#x7528; GitBook &#x53D1;&#x5E03;
        </a>
    </li>
</ul>


                </nav>
            
        
    </div>

    <div class="book-body">
        
            <div class="body-inner">
                
                    

<div class="book-header" role="navigation">
    

    <!-- Title -->
    <h1>
        <i class="fa fa-circle-o-notch fa-spin"></i>
        <a href="..">Read processing algorithms</a>
    </h1>
</div>




                    <div class="page-wrapper" tabindex="-1" role="main">
                        <div class="page-inner">
                            
<div class="search-plus" id="book-search-results">
    <div class="search-noresults">
    
                                <section class="normal markdown-section">
                                
                                <div id="anchor-navigation-ex-navbar"><i class="fa fa-navicon"></i><ul><li><span class="title-icon "></span><a href="#read-processing-algorithms"><b></b>Read processing algorithms</a></li><ul><li><span class="title-icon "></span><a href="#alignment--annotation"><b></b>Alignment &amp; Annotation</a></li><ul><li><span class="title-icon "></span><a href="#cid-mapping"><b></b>CID mapping</a></li><li><span class="title-icon "></span><a href="#rna-filtering"><b></b>RNA filtering</a></li><li><span class="title-icon "></span><a href="#mid-filtering"><b></b>MID filtering</a></li><li><span class="title-icon "></span><a href="#genome-alignment"><b></b>Genome Alignment</a></li><li><span class="title-icon "></span><a href="#annotation-to-transcriptome"><b></b>Annotation to transcriptome</a></li><li><span class="title-icon "></span><a href="#mid-correction--counting"><b></b>MID correction &amp; counting</a></li><li><span class="title-icon "></span><a href="#pid-mapping"><b></b>PID mapping</a></li></ul><li><span class="title-icon "></span><a href="#secondary-analysis"><b></b>Secondary analysis</a></li><ul><li><span class="title-icon "></span><a href="#dimensionality-reduction-"><b></b>Dimensionality reduction </a></li><li><span class="title-icon "></span><a href="#clustering-"><b></b>Clustering </a></li><li><span class="title-icon "></span><a href="#differential-expression-"><b></b>Differential expression </a></li><li><span class="title-icon "></span><a href="#proteome--transcriptome-joint-analysis"><b></b>Proteome &amp; Transcriptome joint analysis</a></li><li><span class="title-icon "></span><a href="#proteome-background-removal"><b></b>Proteome background removal</a></li></ul><li><span class="title-icon "></span><a href="#references"><b></b>References</a></li></ul></ul></div><a href="#read-processing-algorithms" id="anchorNavigationExGoTop"><i class="fa fa-arrow-up"></i></a><h1 id="read-processing-algorithms"><a name="read-processing-algorithms" class="anchor-navigation-ex-anchor" href="#read-processing-algorithms"><i class="fa fa-link" aria-hidden="true"></i></a>Read processing algorithms</h1>
<p><em><mark style="color:purple;">*The interpretation of the Algorithm part is only available in EN.</mark></em></p>
<p>Gene Expression Count during <code>SAW count</code>, for the FASTQ data from the Stereo-seq sequencing platform, includes the following analysis stages:</p>
<figure><a href="../img/assets/Work_with_reads.png" data-lightbox="63e0633e-3a24-4509-bcea-475b6dc6be7c" data-title><img src="../img/assets/Work_with_reads.png" alt></a><figcaption></figcaption></figure>

<p>Regarding the key steps of read alignment and annotation, a flowchart is used to illustrate.</p>
<figure><a href="../img/assets/Workflow_of_Alignment&amp;Annotation.png" data-lightbox="63e0633e-3a24-4509-bcea-475b6dc6be7c" data-title><img src="../img/assets/Workflow_of_Alignment&amp;Annotation.png" alt></a><figcaption></figcaption></figure>

<h2 id="alignment--annotation"><a name="alignment--annotation" class="anchor-navigation-ex-anchor" href="#alignment--annotation"><i class="fa fa-link" aria-hidden="true"></i></a>Alignment &amp; Annotation</h2>
<h3 id="cid-mapping"><a name="cid-mapping" class="anchor-navigation-ex-anchor" href="#cid-mapping"><i class="fa fa-link" aria-hidden="true"></i></a>CID mapping</h3>
<p>CID mapping requires FASTQs and a chip mask file, recording position information for sequencing reads.</p>
<p>Check the amount of Ns in Coordinate IDs:</p>
<ul>
<li>If there is 1 N base in CID, the N base will be replaced by <code>A/T/C/G</code>.</li>
<li>CIDs without N bases will be directly poured into the match pool.</li>
<li>CIDs with more than one N base will be discarded.</li>
</ul>
<p>In the absense of an N base, if a CID does not match to any positions, each base of the CID will replaced by the other three types iterately until a successful match. After the mapping, only the unique CID match will be retained for subsequent steps.</p>
<figure><a href="../img/assets/CID_mapping.png" data-lightbox="63e0633e-3a24-4509-bcea-475b6dc6be7c" data-title><img src="../img/assets/CID_mapping.png" alt></a><figcaption></figcaption></figure>

<h3 id="rna-filtering"><a name="rna-filtering" class="anchor-navigation-ex-anchor" href="#rna-filtering"><i class="fa fa-link" aria-hidden="true"></i></a>RNA filtering</h3>
<p>Before genome alignment, it is necessary to confirm that the reads entering the next step are cDNA sequences. Ideally, this part of the data only contains cDNA fragments, but there may be cases where the fragmented cDNA is too short or some non-cDNA fragments are detected.</p>
<p>Reads will be discarded if any of the following conditions are triggered:</p>
<ul>
<li>with a length of less than 30 after cutting out adapter sequences,</li>
<li>mapped to DNB sequences,</li>
<li>with a length of less than 30 after cutting out the poly-A sequence.</li>
</ul>
<p>The above three are collectively known as &quot;non-relevant short reads&quot;.</p>
<h3 id="mid-filtering"><a name="mid-filtering" class="anchor-navigation-ex-anchor" href="#mid-filtering"><i class="fa fa-link" aria-hidden="true"></i></a>MID filtering</h3>
<p>MID sequences will be filtered out if they match any of the following:</p>
<ul>
<li>having more than one base with quality &lt;= Q10,</li>
<li>having N bases &gt;= 1.</li>
</ul>
<h3 id="genome-alignment"><a name="genome-alignment" class="anchor-navigation-ex-anchor" href="#genome-alignment"><i class="fa fa-link" aria-hidden="true"></i></a>Genome Alignment</h3>
<p>To align reads to the genome with splicing awareness, SAW utilizes the <a href="https://github.com/alexdobin/STAR" target="_blank">STAR</a> aligner. Using a transcript annotation GTF/GFF file, the system classifies reads as exonic, intronic, or intergenic ones based on their alignment. If a read shares at least 50% of its overlap with an exon, it is considered exonic <code>XF:i:0</code>. If a sequence does not meet the criteria for exon but crosses an intron, it is called intronic <code>XF:i:1</code>.</p>
<p>The exonic locus is given priority, and the read is deemed to be reliably mapped to the exonic locus with a MAPQ of 255 for reads that align to one exonic locus but also to one or more non-exonic loci.</p>
<h3 id="annotation-to-transcriptome"><a name="annotation-to-transcriptome" class="anchor-navigation-ex-anchor" href="#annotation-to-transcriptome"><i class="fa fa-link" aria-hidden="true"></i></a>Annotation to transcriptome</h3>
<p>In SAW, reads mapped to exonic and intronic regions are to be annotated, using the transcriptomic annotation file.</p>
<h4 id="remove-rrna"><a name="remove-rrna" class="anchor-navigation-ex-anchor" href="#remove-rrna"><i class="fa fa-link" aria-hidden="true"></i></a>Remove rRNA</h4>
<p>The preconditions for rRNA filtration are:</p>
<ul>
<li>switching <code>--rRNA-remove</code> on when running<code>SAW count</code>,</li>
<li>addition of rRNA information to the index files using <a href="../shi-yong-jiao-cheng/preparation-of-reference.html#for-star"><code>SAW makeRef</code></a>.</li>
</ul>
<p>Reads mapped to rRNA regions will have a tag of <code>XF:i:3</code>, and they will not move on to the next step.</p>
<h4 id="transcriptome"><a name="transcriptome" class="anchor-navigation-ex-anchor" href="#transcriptome"><i class="fa fa-link" aria-hidden="true"></i></a>Transcriptome</h4>
<p>SAW estimates whether there is a successful annotating record based on the position of each read mapped to the genome and whether the alignment is located in a gene region.</p>
<ul>
<li>When the alignment direction of the read on the genome is consistent with that of a gene and overlaps with the gene &gt;= 50% of the mapped length, the read will be considered to be annotated with the gene.</li>
<li>If a read is annotated to multiple genes, the one with the longest overlap length with the gene is selected as the final result of the read. When the overlap lengths are the same, the read is considered to be annotated unsuccessfully.</li>
</ul>
<figure><a href="../img/assets/Annotation.png" data-lightbox="63e0633e-3a24-4509-bcea-475b6dc6be7c" data-title><img src="../img/assets/Annotation.png" alt></a><figcaption></figcaption></figure>

<p>Annotating types are classified according to the following rules:</p>
<ul>
<li>If the overlap between read and exon is &gt;=50%, the read is considered to originate from the exonic region of the gene, otherwise, it originates from the intron region.</li>
<li>Reads overlapped with genes but have inconsistent directions are classified as antisense <code>XF:i:4</code>.</li>
<li>Reads that can be aligned to the genome but cannot be classified as gene or antisense are considered to be intergenic <code>XF:i:2</code>.</li>
</ul>
<h3 id="mid-correction--counting"><a name="mid-correction--counting" class="anchor-navigation-ex-anchor" href="#mid-correction--counting"><i class="fa fa-link" aria-hidden="true"></i></a>MID correction &amp; counting</h3>
<p>At a position, if the MID counts of a certain gene &gt;= 5, they will be corrected. After sorting the MID counts of a specific gene in descending order, the numerical values are corrected following the fundamental principle that allows a Hamming distance of 1. After correction, a tag of <code>UB:Z:XXXXX</code> will be added to the record.</p>
<figure><a href="../img/assets/MID_correction.png" data-lightbox="63e0633e-3a24-4509-bcea-475b6dc6be7c" data-title><img src="../img/assets/MID_correction.png" alt></a><figcaption></figcaption></figure>

<h3 id="pid-mapping"><a name="pid-mapping" class="anchor-navigation-ex-anchor" href="#pid-mapping"><i class="fa fa-link" aria-hidden="true"></i></a>PID mapping</h3>
<p>Check the number of <code>N</code> in 15bp ADT reads.</p>
<ol>
<li>If the number <code>N</code> is greater than 1, discard the read;</li>
<li>If the number  <code>N</code> is 1 and <code>pidMismatch &gt; 0</code>, replace <code>N</code> with <code>A/T/C/G</code>, and then compare with the protein panel respectively. If the query is successful, note <code>CNT = 1</code>. After the iterative query is completed, if <code>CNT = 1</code>, perform <code>pid_mapped_mismatch_reads++.</code> If <code>CNT != 1</code>, discard the read directly.</li>
<li>If the number  <code>N</code> is 0 and <code>pidMismatch = 1</code>, replace any one position of PID with the other three bases for the query. If the number of successful queries is 1, note <code>CNT=1</code> and perform <code>pid_mapped_mismatch_reads++</code>. If the number of successful queries exceeds 1, note <code>CNT != 1</code>, and discard the read.</li>
</ol>
<h2 id="secondary-analysis"><a name="secondary-analysis" class="anchor-navigation-ex-anchor" href="#secondary-analysis"><i class="fa fa-link" aria-hidden="true"></i></a>Secondary analysis</h2>
<p>The secondary analysis, in <code>SAW count</code> and <code>SAW realign</code>, are performed using Stereopy. The procedure includes 6 main parts:</p>
<ul>
<li>Preprocess gene expression data from the tissue coverage region (normalize, logarithmize, identify highly variable genes, and scale each gene).</li>
<li>Reduce the dimensionality of the data by running PCA.</li>
<li>Compute the neighborhood graph and embed the neighborhood graph using UMAP.</li>
<li>Clustering by Leiden algorithm.</li>
<li>Perform differential expression analysis, using the t-test to find marker features.</li>
<li>Output analysis result file in AnnData H5AD and marker feature CSV.</li>
</ul>
<h3 id="dimensionality-reduction-"><a name="dimensionality-reduction-" class="anchor-navigation-ex-anchor" href="#dimensionality-reduction-"><i class="fa fa-link" aria-hidden="true"></i></a>Dimensionality reduction <a href="#dimen-reduction" id="dimen-reduction"></a></h3>
<p><strong>PCA</strong>: Spatiotemporal transcriptome data often contain thousands of genes, making the data high-dimensional and challenging to analyze. Principal Components Analysis (PCA) reduces dimensionality by transforming the data into a set of principal components (PCs) that capture the most significant variance.</p>
<p><strong>UMAP</strong>: SAW supports Uniform Manifold Approximation and Projection (UMAP), which is a popular non-linear dimensionality reduction technique that is particularly useful for visualizing high-dimensional data in a low-dimensional space. In the context of spatiotemporal transcriptome data analysis, it is beneficial to capture both local and global structures as it facilitates the interpretation of complex datasets. UMAP coordinates are available in the pipeline output H5AD, as well as being displayed in the HTML report and StereoMap.</p>
<h3 id="clustering-"><a name="clustering-" class="anchor-navigation-ex-anchor" href="#clustering-"><i class="fa fa-link" aria-hidden="true"></i></a>Clustering <a href="#clustering" id="clustering"></a></h3>
<p>SAW uses Leiden algorithm for clustering spots by expression similarity, operating in the PCA representation.</p>
<p><strong>Leiden:</strong> an algorithm for network community detection, which is particularly important in bioinformatics analysis, especially in the analysis of spatial transcriptomic datasets. It is an enhanced version of Louvain&apos;s method, developed to address its potential instability and provide higher resolution and a more robust community structure. In bioinformatics, the primary roles and benefits of Leiden algorithm can be summarized as follows:</p>
<ul>
<li>Improved resolution and stability: the algorithm can effectively identify smaller community structures, which is crucial for understanding cellular heterogeneity and subtle changes in complex biological processes. Compared with Louvain&apos;s method, Leiden can offer more consistent and stable results across multiple runs.</li>
<li>Efficient cell type identification: in the analysis of spatial transcriptomic sequencing data, Leiden clustering can assist researchers in identifying various cell types or states from thousands of cells. By constructing a network of cell similarities, Leiden algorithm can cluster similar cells together to form distinct groups, which typically correspond to various biological functions or cell types.</li>
<li>Powerful community structure detection: Leiden algorithm ensures the quality of clustering by optimizing a modularity function, which helps reveal complex biological patterns and potential regulatory networks in samples. It is particularly important for understanding the interactions and signal transduction pathways between cells.</li>
<li>Applicable to large-scale datasets: with the advancement of single-cell sequencing technology, the size of datasets is increasing. The Leiden algorithm&apos;s computational efficiency allows it to handle large-scale datasets while keeping computational costs low.</li>
<li>Flexibility and scalability: it is not only applicable to spatial transcriptomic data but can also be extended to other types of bioinformatic data, such as protein interaction networks, metabolic networks, etc.</li>
</ul>
<h3 id="differential-expression-"><a name="differential-expression-" class="anchor-navigation-ex-anchor" href="#differential-expression-"><i class="fa fa-link" aria-hidden="true"></i></a>Differential expression <a href="#diff-exp" id="diff-exp"></a></h3>
<p>To identify genes whose expression is specific to each cluster for each gene and each cluster, whether the in-cluster mean differs from the out-of-cluster mean.</p>
<p><strong>t-test:</strong> the method used in the <code>find_marker_genes</code> of Stereopy package is based on classical statistical principles to identify differentially expressed genes between groups of cells.</p>
<ul>
<li>Grouping cells: cells are grouped based on a categorical variable, such as cluster labels obtained from the Leiden clustering algorithm. For example, cells in cluster A are compared to cells in cluster B.</li>
<li>Expression levels: for each gene, the expression levels are compared between the two groups of cells.</li>
<li>Calculating the t-statistic: the t-statistic is calculated using the formula:</li>
</ul>
<figure><a href="../img/assets/image%20(5).png" data-lightbox="63e0633e-3a24-4509-bcea-475b6dc6be7c" data-title><img src="../img/assets/image (5).png" alt></a><figcaption></figcaption></figure>

<ul>
<li>Degrees of freedom: the degrees of freedom (df) for the test are calculated, often using a formula that takes into account the sample sizes and variances of both groups.</li>
<li>P-value calculation: the t-statistic is then compared to the t-distribution with the calculated degrees of freedom to obtain a p-value. This p-value indicates the probability that the observed difference in means could have occurred by chance.</li>
<li>Multiple testing correction: given that thousands of genes are tested simultaneously, multiple testing correction methods (such as Benjamini-Hochberg) are applied to control the false discovery rate (FDR).</li>
</ul>
<p>The t-test method in <code>find_marker_genes</code> leverages the principles of hypothesis testing to identify genes that are significantly differentially expressed between predefined groups of cells. This method provides a straightforward and statistically sound approach to uncovering key marker genes that distinguish between cell types or conditions, facilitating deeper insights into the underlying biology of the dataset.</p>
<h3 id="proteome--transcriptome-joint-analysis"><a name="proteome--transcriptome-joint-analysis" class="anchor-navigation-ex-anchor" href="#proteome--transcriptome-joint-analysis"><i class="fa fa-link" aria-hidden="true"></i></a>Proteome &amp; Transcriptome joint analysis</h3>
<p>Combining gene and protein data allows us to gain a more comprehensive understanding of cellular identity, as these two perspectives offer different insights into the molecular state. However, it is important to consider the unique sources of technical bias and noise associated with each data type. Specifically, protein data is prone to inherent technical background noise caused by non-specifically bound antibodies. To address these challenges, we utilize Total Variational Inference (totalVI)<a href="gene-expression-algorithms.html#references">[1]</a>, which employs a deep generative model to learn a joint probabilistic representation of paired gene and protein measurements. This approach effectively accounts for the distinct noise and technical biases of each modality. Within this framework, our tool, SAW, leverages totalVI to generate a joint latent representation of spots (or cells), produce denoised gene and protein data, and compute differential expression of gene and protein. </p>
<p>To initiate the totalVI model, SAW inputs matrices of gene and protein unique molecular identifier counts (GEF file) for spots or cells. The model then generates a joint low-dimensional latent space. For cluster identification, SAW employs the Leiden algorithm and visualizes the clusters using UMAP. Differential expression testing (DE test) is conducted using totalVI, which estimates a distribution over the log fold change (LFC) of expression between two sets of spots (or cells). This distribution is utilized to quantify the strength of evidence supporting a hypothesis of differential expression, employing Bayes factors. SAW performs a one-vs-all DE test to determine gene and protein markers for each cluster, respectively, and stores the results in a single dataframe in CSV format. To filter the results, we retain features with a Bayes factor on the natural log scale above a specified threshold (&gt;1 for gene markers, &gt;0.7 for protein markers), as well as genes with an expression proportion greater than 10% in the cluster of interest. The denoised gene and protein data are produced in the H5MU format, allowing for visualization of gene and protein markers within each cluster using a heatmap.</p>
<h3 id="proteome-background-removal"><a name="proteome-background-removal" class="anchor-navigation-ex-anchor" href="#proteome-background-removal"><i class="fa fa-link" aria-hidden="true"></i></a>Proteome background removal</h3>
<p>There is non-specificity in protein antibody binding to cell surface antigens in tissues. Therefore, the antibody protein signal intensity detected in the tissue includes two situations:</p>
<ul>
<li>Protein positive signal is specific binding;</li>
<li>Protein negative signal is non-specific binding. </li>
</ul>
<p>Therefore, it is necessary to dynamically determine a reasonable threshold to distinguish protein positive signals from negative signals. Different antibody protein thresholds are independent of each other and do not affect each other, and the threshold must meet the following conditions:</p>
<ul>
<li>The intra-class variance of the protein positive signal set and the negative signal set is the smallest</li>
<li>The inter-class variance of the protein positive signal set and the negative signal set is the largest</li>
</ul>
<p>In order to complete the above conditions, it is necessary to perform&#xFF1A;</p>
<ol>
<li>Remove the duplicates of the protein signal intensity and sort it from small to large, noted as <code>(m1, m2, m3, m4, ..., mn)</code>, and then count the frequency of occurrence of each signal intensity <code>m</code>, noted as <code>(f1, f2, f3, f4, ..., fn)</code>.</li>
<li>Set the initial threshold <code>t</code>, and <code>t</code> satisfies <code>m1 &lt; t &lt; mn</code>. Then take <code>t</code> as the threshold, classify those greater than <code>t</code> into one category, note them as<code>C1</code>. Those less than or equal to <code>t</code> are classified into one category, noted as <code>C2</code>. Then calculate the inter-class and intra-class variances of <code>C1</code> and <code>C2</code>. Since the sum of the intra-class and inter-class variances is a constant when the inter-class variance is the largest, the intra-class variance is the smallest, so only record the inter-class variance corresponding to the <code>t</code> value.</li>
<li>Iterate the <code>t</code> value and record the inter-class variance corresponding to each <code>t</code>. After the iteration is completed, take the <code>t</code> value corresponding to the maximum inter-class variance as the reasonable threshold to distinguish the positive signal from the negative signal of the protein.</li>
<li>According to the determined threshold, perform the following operations: A value greater than the threshold is considered a specific binding signal. A value less than or equal to the threshold is considered a non-specific binding signal. Then filter out the bin or cell of the non-specific binding signal to achieve background removal.</li>
</ol>
<h2 id="references"><a name="references" class="anchor-navigation-ex-anchor" href="#references"><i class="fa fa-link" aria-hidden="true"></i></a>References</h2>
<ol>
<li>Gayoso, A., Steier, Z., Lopez, R., Regier, J., Nazor, K. L., Streets, A., &amp; Yosef, N. (2021). Joint probabilistic modeling of single-cell multi-omic data with totalVI. Nature Methods, 18(3), 272-282.</li>
</ol>
<footer class="page-footer"><span class="copyright">&#xA9; 2025 STOmics Tech. All rights reserved.</span><span class="footer-modification">Modified: 
2025-04-09 15:29:46
</span></footer><script>document.addEventListener("DOMContentLoaded", function() {lightbox.option({"resizeDuration":200,"wrapAround":true});})</script>
                                
                                </section>
                            
    </div>
    <div class="search-results">
        <div class="has-results">
            
            <h1 class="search-results-title"><span class="search-results-count"></span> results matching &quot;<span class="search-query"></span>&quot;</h1>
            <ul class="search-results-list"></ul>
            
        </div>
        <div class="no-results">
            
            <h1 class="search-results-title">No results matching &quot;<span class="search-query"></span>&quot;</h1>
            
        </div>
    </div>
</div>

                        </div>
                    </div>
                
            </div>

            
                
                <a href="./" class="navigation navigation-prev " aria-label="Previous page: &#x7B97;&#x6CD5;&#x539F;&#x7406;">
                    <i class="fa fa-angle-left"></i>
                </a>
                
                
                <a href="image-processing-algorithms.html" class="navigation navigation-next " aria-label="Next page: Image processing algorithms">
                    <i class="fa fa-angle-right"></i>
                </a>
                
            
        
    </div>

    <script>
        var gitbook = gitbook || [];
        gitbook.push(function() {
            gitbook.page.hasChanged({"page":{"title":"Read processing algorithms","level":"1.6.1","depth":2,"next":{"title":"Image processing algorithms","level":"1.6.2","depth":2,"path":"algorithms/image-processing-algorithms.md","ref":"algorithms/image-processing-algorithms.md","articles":[]},"previous":{"title":"算法原理","level":"1.6","depth":1,"path":"algorithms/README.md","ref":"algorithms/README.md","articles":[{"title":"Read processing algorithms","level":"1.6.1","depth":2,"path":"algorithms/gene-expression-algorithms.md","ref":"algorithms/gene-expression-algorithms.md","articles":[]},{"title":"Image processing algorithms","level":"1.6.2","depth":2,"path":"algorithms/image-processing-algorithms.md","ref":"algorithms/image-processing-algorithms.md","articles":[]}]},"dir":"ltr"},"config":{"plugins":["expandable-chapters-small","anchor-navigation-ex","logo","-insert-logo","tbfed-pagefooter","hide-element","-code-optimize","katex","hints","tabs","lightbox","-lunr","-search","search-plus","add-index-html","anchor-to-full-links"],"root":".","styles":{},"pluginsConfig":{"tbfed-pagefooter":{"copyright":"© 2025 STOmics Tech.","modify_label":"Modified: ","modify_format":"YYYY-MM-DD HH:mm:ss"},"logo":{"logo":"data:image/png;base64,AAAAGGZ0eXBhdmlmAAAAAG1pZjFtaWFmAAAA0m1ldGEAAAAAAAAAIWhkbHIAAAAAAAAAAHBpY3QAAAAAAAAAAAAAAAAAAAAADnBpdG0AAAAAAAEAAAAeaWxvYwAAAABEAAABAAEAAAABAAAA8gAABgAAAAAjaWluZgAAAAAAAQAAABVpbmZlAgAAAAABAABhdjAxAAAAAFZpcHJwAAAAOGlwY28AAAAUaXNwZQAAAAAAAAAgAAAALQAAAAxhdjFDgT9AAAAAABBwaXhpAAAAAAMKCgoAAAAWaXBtYQAAAAAAAAABAAEDAYIDAAAGCG1kYXQSAAoJP9F/YXgIaDaAMvALZAQYAAQAAgIAAAAAAAAAEAAAAAVACJZH5jEfcnHq5GfVOqfqAcroK/CW/n75YIMe8ZfAjZNFE/6UrpMmpDP/er+E0q/ceflnZGp4dyYbk4GCBlbhdKT3x+D/AIaisuTFd8nYmo+ez1m+B7mjO9pucuLd4J09fZofEcRZ1N8yXb1uByDJRlxGRNCsg51xqZYiGzKHoIgR4ChwET50S4FSgHUScr3K3YmkXZjZK1Sle8EZjq6uDLYWGw4NvJ/pvNMkd/WmxjuUTT7pfVtNIGjG28GzYzHlf3vaTaGjRUC91DO9vrJT51k9bZZg8h2LiVbz7yNTOZcjlBLZuokz15s00uLVyWobV2CGySmbuxBbAsKxItjkVbSHo6ojiJbY5bPJvksg0SaezQgA1EgcN96PcNp/fjOFRBtevV/O+rjq23jHY8C9n/YSCdXnGsWJNHdxleLb40bjcNXoXncQADGMuX4XOfKPzQ3L6GO6nlgQBXZyDdvAAbF778o6nSAV86cHKEWnD3tckPflzq+6jvx4oCjObYt0X1Ro63jOf55vCFAlxB8FJNtvuok/MW0bSKvp+0LTGESD/2lmWnH+IvOX5hbFmtTw5mgGW0eKFHyZXqg00hWU4wfzKPxpRwvxAUZMaiFlEudhK9HjhFYdgz/1ShJlEUXzbR+U6nabVjctZv4ZQeECNYZ3p4pAGxtsSlxZBYQ4ARr8zqcDQjsYk+rr1KdIQ14PnyCIe0imW/rJ/at5+YUvZ6Z4FApMELY7udGJJ5HuVaPCr22ij/h9mvhjgSMMfCesXRgyxDAWg9Aa+lx/LxDWkXLxHot/ByjPyng7o28F0TwpKuvCU8x8MxSuy9o89zSiGVR2nsz7EEWJvFEeZKRMn9j30VIwqcF4D/tWMO3puIoih0v/CvKusdf7v0qTb58ZaIGvRrTJSLiosclEEtiORGa0wihqVOcL7O3GL2rIRP/Pcv1BSDYXhybnLZ+HbhjuBO2ulzWv3j+0GWtGdtmij8t9ivPRKOC7RYXLQ99mYd621NSxNWbNJaP1XYbLFkbATt+nBDr+ypqyy+ARzVlE9ywft4Zj683T2knXPPhbo0X9I70mu1LIAsFCr9IwunPhTV//P4W4w8S1Ks9IB7NeBdiWCCtW1KWtIuFaIPvx+VbZkxyad+Nqh4iFTKBYIfGffIbiEQiBaJaxB+uIC8/TylgtlADj00ipv0HgSHv3v2rna/hOViWRbA9kars4YRbv8GF4i4delTz8Myp9DsZQv/owkRFJkk5KHrv25DT0CrW2FxlBIuJQuSHak+OCRtl+RnJwNd+0wDZ09WRFCkbt6qoCbVumBWzBM/0afWjK8QukftmXeX93+lbWGWrOova/TResZhzunNubiTDOH6WgLh/g8v3YUg1S5V2GzESjFUtpQBZFiQVn6a+beAxr3kQINzRSASF1nZZjdiEmN7rXfuhoqkBJV6uARAJBDfPNARGsjTgivGjqjHdjeZHNowz9apljGORFaJJBbE0zTjcd7NUmAiSd9dIXG8cAmiBmmU1fUbaupjLvmwgxdI9ZxfpP3h7WygcJklKiA+Unx+LJT20S7nW1nO/HiqEKOAnhKKEvpWo8NWVVJ1l1HXanmZRnuA+DkpOr4ahp63B1LxmbWnEP8IzsZtbQWG1+yQUfwN4CYnnPZB34ljqnPbGb+ReVnQ/B552w6HkZFlnSYxvuVvAg6LnzslXN9UZZreNj6xTh9AAsznkSmpb7//5b4QpSnH9N948K/lP3QND+g6Co5XKGmAfUS+NnBs4/AzZQg+3n/VjAIHA5YIMMduPmXmlwzDiQOZC2pSYfpMUqIScTuITS35U5C8lq86m1jGD1ZSZRse4fJdFbPKs+TldyLZyqXub516/e1dRG47gRwSUoWfcjoG486GAZTabIiCOxD9Pq13HejOuZG+j+/0zALlXdSctzuOMx1XJMaYHpJ06kZmkCGN+JXeuDjk32LVGFUn/WY7XCw1tvJQBykA6dQTtjhbrhALglaNHrFiw=","title":"SAW 用户手册 8.1 ","titleStyle":"font-weight:600"},"hints":{"danger":"fa fa-exclamation-circle","info":"fa fa-info-circle","tip":"fa fa-mortar-board","working":"fa fa-wrench"},"hide-element":{"elements":[".gitbook-link"]},"katex":{},"add-index-html":{"myConfigKey":"it's the default value","elements":["a"]},"fontsettings":{"theme":"white","family":"sans","size":2},"highlight":{},"anchor-navigation-ex":{"associatedWithSummary":true,"float":{"floatIcon":"fa fa-navicon","level1Icon":"fa fa-hand-o-right","level2Icon":"fa fa-hand-o-right","level3Icon":"fa fa-hand-o-right","showLevelIcon":false},"mode":"float","multipleH1":true,"pageTop":{"level1Icon":"fa fa-hand-o-right","level2Icon":"fa fa-hand-o-right","level3Icon":"fa fa-hand-o-right","showLevelIcon":false},"printLog":true,"showGoTop":true,"showLevel":false},"anchor-to-full-links":{"myConfigKey":"it's the default value","elements":["a"]},"lightbox":{"jquery":true,"includeJQuery":true,"sameUuid":true,"options":{"resizeDuration":200,"wrapAround":true}},"expandable-chapters-small":{},"sharing":{"facebook":false,"twitter":false,"google":false,"weibo":false,"instapaper":false,"vk":false,"all":[]},"theme-default":{"styles":{"website":"styles/website.css","pdf":"styles/pdf.css","epub":"styles/epub.css","mobi":"styles/mobi.css","ebook":"styles/ebook.css","print":"styles/print.css"},"showLevel":false},"code-optimize":{"copyButtons":true,"showLines":true},"tabs":{},"search-plus":{}},"theme":"default","author":"","pdf":{"pageBreaksBefore":"/","headerTemplate":null,"paperSize":"a4","margin":{"right":62,"left":62,"top":36,"bottom":36},"fontSize":12,"fontFamily":"Arial","footerTemplate":null,"chapterMark":"pagebreak","pageNumbers":false},"structure":{"langs":"LANGS.md","readme":"README.md","glossary":"GLOSSARY.md","summary":"SUMMARY.md"},"isbn":"","variables":{},"title":"SAW User Manual","language":"zh-hans","links":{"gitbook":false,"sharing":{"google":false,"facebook":false,"twitter":false,"all":false}},"gitbook":"3.2.3","description":null,"extension":null},"file":{"path":"algorithms/gene-expression-algorithms.md","mtime":"2025-04-09T07:29:46.834Z","type":"markdown"},"gitbook":{"version":"3.2.3","time":"2025-04-09T09:29:48.718Z"},"basePath":"..","book":{"language":""}});
        });
    </script>
</div>

        
    <script src="../gitbook/gitbook.js"></script>
    <script src="../gitbook/theme.js"></script>
    
        
        <script src="../gitbook/gitbook-plugin-expandable-chapters-small/expandable-chapters-small.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-hide-element/plugin.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-tabs/bootstrap.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-lightbox/js/lightbox.min.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-search-plus/jquery.mark.min.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-search-plus/search.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-add-index-html/plugin.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-anchor-to-full-links/plugin.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-sharing/buttons.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-fontsettings/fontsettings.js"></script>
        
    

    </body>
</html>

