
<!DOCTYPE HTML>
<html lang="zh-hans" >
    <head>
        <meta charset="UTF-8">
        <meta content="text/html; charset=utf-8" http-equiv="Content-Type">
        <title>Expression matrix format · SAW User Manual</title>
        <meta http-equiv="X-UA-Compatible" content="IE=edge" />
        <meta name="description" content="">
        <meta name="generator" content="GitBook 3.2.3">
        
        
        
    
    <link rel="stylesheet" href="../gitbook/style.css">

    
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-expandable-chapters-small/expandable-chapters-small.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-anchor-navigation-ex/style/plugin.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-tbfed-pagefooter/footer.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-code-optimize/plugin.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-katex/katex.min.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-hints/plugin-hints.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-tabs/bootstrap.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-lightbox/css/lightbox.min.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-search-plus/search.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-highlight/website.css">
                
            
                
                <link rel="stylesheet" href="../gitbook/gitbook-plugin-fontsettings/website.css">
                
            
        

    

    

        
    
    
    <meta name="HandheldFriendly" content="true"/>
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <link rel="apple-touch-icon-precomposed" sizes="152x152" href="../gitbook/images/apple-touch-icon-precomposed-152.png">
    <link rel="shortcut icon" href="../gitbook/images/favicon.ico" type="image/x-icon">

    
    <link rel="next" href="image-related-record-format.html" />
    
    
    <link rel="prev" href="analysis-with-gpu.html" />
    

    </head>
    <body>
        
<div class="book">
    <div class="book-summary">
        
            
<div id="book-search-input" role="search">
    <input type="text" placeholder="输入并搜索" />
</div>

            
                <nav role="navigation">
                


<ul class="summary">
    
    

    

    
        
        
    
        <li class="chapter " data-level="1.1" data-path="../">
            
                <a href="../">
            
                    
                    概览
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.2" data-path="../getting-started/">
            
                <a href="../getting-started/">
            
                    
                    快速开始
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.2.1" data-path="../getting-started/introduction.html">
            
                <a href="../getting-started/introduction.html">
            
                    
                    软件介绍
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.2.2" data-path="../getting-started/glossary.html">
            
                <a href="../getting-started/glossary.html">
            
                    
                    词汇表
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.3" data-path="../download-center.html">
            
                <a href="../download-center.html">
            
                    
                    下载中心
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4" data-path="../analysis/">
            
                <a href="../analysis/">
            
                    
                    运行分析
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.1" data-path="../analysis/inputs/">
            
                <a href="../analysis/inputs/">
            
                    
                    输入文件
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.1.1" data-path="../analysis/inputs/chip-mask.html">
            
                <a href="../analysis/inputs/chip-mask.html">
            
                    
                    芯片 mask
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.1.2" data-path="../analysis/inputs/fastqs.html">
            
                <a href="../analysis/inputs/fastqs.html">
            
                    
                    测序 FASTQ
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.1.3" data-path="../analysis/inputs/images.html">
            
                <a href="../analysis/inputs/images.html">
            
                    
                    显微镜图像
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.4.2" data-path="../analysis/pipelines/">
            
                <a href="../analysis/pipelines/">
            
                    
                    分析流程
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.2.1" data-path="../analysis/pipelines/saw-can-shu-ming-ling.html">
            
                <a href="../analysis/pipelines/saw-can-shu-ming-ling.html">
            
                    
                    SAW 参数命令
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.2.2" data-path="../analysis/pipelines/count/">
            
                <a href="../analysis/pipelines/count/">
            
                    
                    count
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.2.2.1" data-path="../analysis/pipelines/count/stereo-seq-t-ff.html">
            
                <a href="../analysis/pipelines/count/stereo-seq-t-ff.html">
            
                    
                    Stereo-seq T FF
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.2.2.2" data-path="../analysis/pipelines/count/stereo-seq-n-ffpe-v1.0.html">
            
                <a href="../analysis/pipelines/count/stereo-seq-n-ffpe-v1.0.html">
            
                    
                    Stereo-seq N FFPE
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.4.3" data-path="../analysis/outputs/">
            
                <a href="../analysis/outputs/">
            
                    
                    输出结果
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.4.3.1" data-path="../analysis/outputs/count-outputs.html">
            
                <a href="../analysis/outputs/count-outputs.html">
            
                    
                    count 输出结果
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.2" data-path="../analysis/outputs/realign-outputs.html">
            
                <a href="../analysis/outputs/realign-outputs.html">
            
                    
                    realign 输出结果
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.3" data-path="../analysis/outputs/html-report.html">
            
                <a href="../analysis/outputs/html-report.html">
            
                    
                    HTML report
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.4" data-path="../analysis/outputs/bam.html">
            
                <a href="../analysis/outputs/bam.html">
            
                    
                    BAM
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.5" data-path="../analysis/outputs/matrices.html">
            
                <a href="../analysis/outputs/matrices.html">
            
                    
                    Matrices
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.6" data-path="../analysis/outputs/images.html">
            
                <a href="../analysis/outputs/images.html">
            
                    
                    Images
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.4.3.7" data-path="../analysis/outputs/analysis.html">
            
                <a href="../analysis/outputs/analysis.html">
            
                    
                    Analysis
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.5" data-path="../shi-yong-jiao-cheng/">
            
                <a href="../shi-yong-jiao-cheng/">
            
                    
                    使用教程
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.5.1" data-path="../shi-yong-jiao-cheng/preparation-of-reference.html">
            
                <a href="../shi-yong-jiao-cheng/preparation-of-reference.html">
            
                    
                    比对索引构建
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.2" data-path="../shi-yong-jiao-cheng/stereo-seq-ff.html">
            
                <a href="../shi-yong-jiao-cheng/stereo-seq-ff.html">
            
                    
                    Stereo-seq FF 分析
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.3" data-path="../shi-yong-jiao-cheng/stereo-seq-ffpe.html">
            
                <a href="../shi-yong-jiao-cheng/stereo-seq-ffpe.html">
            
                    
                    Stereo-seq FFPE 分析
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.4" data-path="../shi-yong-jiao-cheng/shou-dong-tu-xiang-chu-li.html">
            
                <a href="../shi-yong-jiao-cheng/shou-dong-tu-xiang-chu-li.html">
            
                    
                    手动图像处理
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.5" data-path="../shi-yong-jiao-cheng/secondary-analysis.html">
            
                <a href="../shi-yong-jiao-cheng/secondary-analysis.html">
            
                    
                    数据再分析
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.5.6" data-path="../shi-yong-jiao-cheng/shu-ju-ge-shi-zhuan-huan.html">
            
                <a href="../shi-yong-jiao-cheng/shu-ju-ge-shi-zhuan-huan.html">
            
                    
                    数据格式转换
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.6" data-path="../algorithms/">
            
                <a href="../algorithms/">
            
                    
                    算法原理
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.6.1" data-path="../algorithms/gene-expression-algorithms.html">
            
                <a href="../algorithms/gene-expression-algorithms.html">
            
                    
                    Read processing algorithms
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.6.2" data-path="../algorithms/image-processing-algorithms.html">
            
                <a href="../algorithms/image-processing-algorithms.html">
            
                    
                    Image processing algorithms
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.7" data-path="./">
            
                <a href="./">
            
                    
                    高级设置
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="1.7.1" data-path="analysis-with-gpu.html">
            
                <a href="analysis-with-gpu.html">
            
                    
                    Analysis with GPU
            
                </a>
            

            
        </li>
    
        <li class="chapter active" data-level="1.7.2" data-path="expression-matrix-format.html">
            
                <a href="expression-matrix-format.html">
            
                    
                    Expression matrix format
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.7.3" data-path="image-related-record-format.html">
            
                <a href="image-related-record-format.html">
            
                    
                    Image-related record format
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="1.8" data-path="../cuo-wu-ma.html">
            
                <a href="../cuo-wu-ma.html">
            
                    
                    错误码
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.9" data-path="../release-notes.html">
            
                <a href="../release-notes.html">
            
                    
                    版本说明
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.10" data-path="../shu-ju-ji.html">
            
                <a href="../shu-ju-ji.html">
            
                    
                    数据集
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.11" >
            
                <a target="_blank" href="https://www.stomics.tech/resources/FAQ">
            
                    
                    常见问题
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="1.12" >
            
                <a target="_blank" href="https://www.stomics.tech/service/new-StereoMap.html">
            
                    
                    StereoMap
            
                </a>
            

            
        </li>
    

    

    <li class="divider"></li>

    <li>
        <a href="https://www.gitbook.com" target="blank" class="gitbook-link">
            本书使用 GitBook 发布
        </a>
    </li>
</ul>


                </nav>
            
        
    </div>

    <div class="book-body">
        
            <div class="body-inner">
                
                    

<div class="book-header" role="navigation">
    

    <!-- Title -->
    <h1>
        <i class="fa fa-circle-o-notch fa-spin"></i>
        <a href=".." >Expression matrix format</a>
    </h1>
</div>




                    <div class="page-wrapper" tabindex="-1" role="main">
                        <div class="page-inner">
                            
<div class="search-plus" id="book-search-results">
    <div class="search-noresults">
    
                                <section class="normal markdown-section">
                                
                                <div id="anchor-navigation-ex-navbar"><i class="fa fa-navicon"></i><ul><li><span class="title-icon "></span><a href="#expression-matrix-format"><b></b>Expression matrix format</a></li><ul><li><span class="title-icon "></span><a href="#gene-expression-file-gef"><b></b>Gene Expression File (GEF)</a></li><ul><li><span class="title-icon "></span><a href="#bin-gef"><b></b>Bin GEF</a></li><li><span class="title-icon "></span><a href="#cell-bin-gef"><b></b>Cell Bin GEF</a></li></ul><li><span class="title-icon "></span><a href="#gene-expression-matrix-gem"><b></b>Gene Expression Matrix (GEM)</a></li></ul></ul></div><a href="#expression-matrix-format" id="anchorNavigationExGoTop"><i class="fa fa-arrow-up"></i></a><h1 id="expression-matrix-format"><a name="expression-matrix-format" class="anchor-navigation-ex-anchor" href="#expression-matrix-format"><i class="fa fa-link" aria-hidden="true"></i></a>Expression matrix format</h1>
<p><em><mark style="color:purple;">*The interpretation of the Advanced part is only available in EN.</mark></em></p>
<h2 id="gene-expression-file-gef"><a name="gene-expression-file-gef" class="anchor-navigation-ex-anchor" href="#gene-expression-file-gef"><i class="fa fa-link" aria-hidden="true"></i></a>Gene Expression File (GEF)</h2>
<p>Gene expression file (GEF) is a data management and storage format designed to support multidimensional datasets and high computational efficiency. Stereo-seq analysis workflow generates bin GEF and cellbin GEF files. Bin GEF file format is a hierarchically structured data model that stores one or bin combined gene expression matrices in various bin sizes. Cellbin GEF file format stores expression information within each cell. Each GEF container organizes a collection of spatial gene expression matrices. It includes two primary data objects: Group and Dataset. A dataset is a multidimensional array of data elements. Group object is analogous to a file system directory that organizes datasets and other groups in hierarchies.</p>
<h3 id="bin-gef"><a name="bin-gef" class="anchor-navigation-ex-anchor" href="#bin-gef"><i class="fa fa-link" aria-hidden="true"></i></a>Bin GEF</h3>
<p>The first level of GEF includes four group objects: &quot;geneExp&quot; (required), &quot;wholeExp&quot; (optional), &quot;wholeExpExon&quot; (optional), and &quot;stat&quot; (optional). Group &quot;geneExp&quot; contains groups of gene spatial expression data in one or multiple bin sizes. Group &quot;wholeExp&quot; contains datasets that record expression level and gene type count of each coordinate in one or multiple bin sizes. Group &quot;wholeExpExon&quot; contains datasets that record the exon level of each coordinate in one or multiple bin sizes. Group &quot;stat&quot; saves gene names, total MID count and spatial pattern enrichment score of each gene. &quot;Attributes&quot; of the file record the version of GEF format, software version, and omics information. &quot;Attribute&quot; in each dataset records the key metrics of that dataset. Check the table to get details.</p>
<table>
<thead>
<tr>
<th><em><strong>Attributes</strong></em></th>
<th></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>File Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>version</td>
<td>uint32</td>
<td>2</td>
<td>Gene expression file format version.</td>
</tr>
<tr>
<td>geftool_ver</td>
<td><p>uint32[3]<br></p></td>
<td>1,11,12</td>
<td>Geftool version. It can be used as an individual tool to manipulate GEF files.</td>
</tr>
<tr>
<td>omics</td>
<td>S32</td>
<td>b&apos;Transcriptomics&apos;</td>
<td>Omics name.</td>
</tr>
<tr>
<td>gef_area</td>
<td>float32</td>
<td>4.4410855E10</td>
<td>Tissue or labeled tissue area in square nanometers.</td>
</tr>
<tr>
<td>bin_type</td>
<td>S32</td>
<td>b&apos;bin&apos;</td>
<td>Bin type of the GEF file.</td>
</tr>
<tr>
<td>sn</td>
<td>S32</td>
<td>b&apos;SS200000135TL_D1&apos;</td>
<td>Stereo-seq chip SN</td>
</tr>
<tr>
<td><em><strong>/geneExp/binN/expression:Dataset &quot;expression&quot; is a 1D array which stores coordinates and MID counts of each gene in the bin size of N, aggregated by gene name.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example (bin1)</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>minX</td>
<td>int32</td>
<td>59820</td>
<td>Minimum x coordinate in bin N.</td>
</tr>
<tr>
<td>minY</td>
<td>int32</td>
<td>102086</td>
<td>Minimum y coordinate in bin N.</td>
</tr>
<tr>
<td>maxX</td>
<td>int32</td>
<td>73040</td>
<td>Maximum x coordinate in bin N.</td>
</tr>
<tr>
<td>maxY</td>
<td>int32</td>
<td>120539</td>
<td>Maximum y coordinate in bin N.</td>
</tr>
<tr>
<td>maxExp</td>
<td>uint32</td>
<td>28</td>
<td>Maximum MID count in a spot when the bin size is N. Data type for &quot;maxExp&quot; is dynamically changed for each sample.</td>
</tr>
<tr>
<td>resolution</td>
<td>uint32</td>
<td>500</td>
<td>Physical pitch (nm) between neighbor spots.</td>
</tr>
<tr>
<td><strong>Dataset DataType:compound</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example (bin1)</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>x</td>
<td>int32</td>
<td>71032</td>
<td>x coordinate in bin N.</td>
</tr>
<tr>
<td>y</td>
<td>int32</td>
<td>103180</td>
<td>y coordinate in bin N.</td>
</tr>
<tr>
<td>count</td>
<td>uint8/uint16/uint32</td>
<td>1</td>
<td>MID count at (x, y) when bin size is N. Data type for &quot;count&quot; is consistent with &quot;maxExp&quot; in the &quot;Attributes.&quot;</td>
</tr>
<tr>
<td><em><strong>[optional] /geneExp/binN/exon:Dataset &quot;exon&quot; is a 1D array which stores exon expression of each gene in the bin size of N, aggregated by gene name.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example (bin1)</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>maxExon</td>
<td>int32</td>
<td>21</td>
<td>Max exon expression in binN.</td>
</tr>
<tr>
<td><strong>Dataset DataType:1D array</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example (bin1)</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>count</td>
<td>uint8/uint16/uint32</td>
<td>0</td>
<td>Exon expression in binN at coordinate (x,y), the index is same to the index in the &quot;expression&quot; dataset. Data type for &quot;count&quot; is dynamically changed for each sample.</td>
</tr>
<tr>
<td><strong>/geneExp/binN/gene:Dataset &quot;gene&quot; is a 1D array which stores the gene names, the starting row indexes in dataset &quot;expression&quot;, and row counts.</strong></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset DataType:compound</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example (bin1)</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>geneID</td>
<td>S64</td>
<td>b&apos;ENSMUSG00000000001&apos;</td>
<td>Gene ID.</td>
</tr>
<tr>
<td>geneName</td>
<td>S64</td>
<td>b&apos;Gm16045&apos;</td>
<td>Gene name.</td>
</tr>
<tr>
<td>offset</td>
<td>uint32</td>
<td>21</td>
<td>The starting row index in dataset &quot;expression&quot; for the gene.In this example, the gene expression data for gene &quot;Gm16045&quot; starts from row 21 in the dataset &quot;expression.&quot;</td>
</tr>
<tr>
<td>count</td>
<td>uint32</td>
<td>2</td>
<td>Row count.In this example, expression data for gene &quot;Gm16045&quot; is recorded in row 21 and 22 (2 rows) in the dataset &quot;expression.&quot;</td>
</tr>
<tr>
<td><em><strong>[optional] /wholeExp/binN:Dataset &quot;binN&quot; is a 2D array (matrix) which stores the MID count and gene type count at each spot.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example (bin1)</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>number</td>
<td>uint64</td>
<td>22879557</td>
<td>Number of non-zero spots in the dense matrix.</td>
</tr>
<tr>
<td>minX</td>
<td>int32</td>
<td>59820</td>
<td>Minimum x coordinate in bin N.</td>
</tr>
<tr>
<td>lenX</td>
<td>int32</td>
<td>13221</td>
<td>Length of x.</td>
</tr>
<tr>
<td>minY</td>
<td>int32</td>
<td>102086</td>
<td>Minimum y coordinate in bin N.</td>
</tr>
<tr>
<td>lenY</td>
<td>int32</td>
<td>18454</td>
<td>Length of y.</td>
</tr>
<tr>
<td>maxMID</td>
<td>uint32</td>
<td>2155</td>
<td>Maximum MID count in a spot.</td>
</tr>
<tr>
<td>maxGene</td>
<td>uint32</td>
<td>846</td>
<td>Maximum gene type count in a spot.</td>
</tr>
<tr>
<td>resolution</td>
<td>uint32</td>
<td>500</td>
<td>Pitch (nm) between neighbor spots.</td>
</tr>
<tr>
<td><strong>Dataset DataType: 2D array (X&#x2179;Y), compound</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example (bin1)</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>MIDcount</td>
<td>uint8/uint16/uint32</td>
<td>1</td>
<td>MID count in the spot. The spot coordinate can be identified from the row and column index of the 2D matrix plus the &quot;minX&quot; and &quot;minY&quot; specified in the attributes. Data type for &quot;MIDcount&quot; is dynamically changed for each sample.</td>
</tr>
<tr>
<td>genecount</td>
<td>uint16</td>
<td>1</td>
<td>Gene count in the spot. The spot coordinate can be identified from &quot;Attributes&quot; and the indexes of the 2D array.</td>
</tr>
<tr>
<td><em><strong>[optional] /wholeExpExon/binN:Dataset &quot;binN&quot; in &quot;/wholeExpExon/&quot; Group is a 2D array (matrix) which stores the exon expression count at each spot.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example (bin1)</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>maxExon</td>
<td>uint32</td>
<td>21</td>
<td>Maximum exon expression count in a spot when the bin size is N.</td>
</tr>
<tr>
<td><strong>Dataset DataType: 2D array</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example (bin1)</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>MIDcount</td>
<td>uint8/uint16/uint32</td>
<td>0</td>
<td>MID count in the spot. The spot coordinate can be identified from the row and column index of the 2D matrix plus the &quot;minX&quot; and &quot;minY&quot; specified in the attributes. Data type for &quot;MIDcount&quot; is dynamically changed for each sample.</td>
</tr>
<tr>
<td><em><strong>[optional] /stat/gene:Dataset &quot;gene&quot; is a 1D array which stores the MID count and spatial pattern enrichment score (E10) of each gene. The array is order by the MID count in descending order.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>maxE10</td>
<td>float32</td>
<td>65.53</td>
<td>Maximum E10 score.</td>
</tr>
<tr>
<td>minE10</td>
<td>float32</td>
<td>0.</td>
<td>Minimum E10 score.</td>
</tr>
<tr>
<td>cutoff</td>
<td>float32</td>
<td>0.1</td>
<td>Threshold for filtering spots that will be used for computing E10.In this example, 0.1 means that the spots whose MID count is in the top 10% are used for calculating the spatial enrichment score.</td>
</tr>
<tr>
<td><strong>Dataset DataType:compound</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>geneID</td>
<td>S64</td>
<td>b&apos;ENSMUSG00000000001&apos;</td>
<td>Gene ID.</td>
</tr>
<tr>
<td>geneName</td>
<td>S64</td>
<td>b&apos;Ptgds&apos;</td>
<td>Gene name.</td>
</tr>
<tr>
<td>MIDcount</td>
<td>uint32</td>
<td>229502</td>
<td>MID count for the gene.</td>
</tr>
<tr>
<td>E10</td>
<td>float32</td>
<td>65.53</td>
<td>The spatial pattern enrichment score (E10) for the gene.</td>
</tr>
</tbody>
</table>
<h3 id="cell-bin-gef"><a name="cell-bin-gef" class="anchor-navigation-ex-anchor" href="#cell-bin-gef"><i class="fa fa-link" aria-hidden="true"></i></a>Cell Bin GEF</h3>
<p>The first layer of Cell Bin GEF contains one required group &quot;cellBin&quot; and multiple optional datasets. The second layer &quot;codedCellBlock&quot; is optional, which stores precomputed data used in the rendering of <strong>StereoMap</strong>. &quot;Attributes&quot; of the file record the version of GEF format, software version, and omics information. &quot;Attribute&quot; in each dataset records the key metrics of that dataset. Check the table to get more details.</p>
<table>
<thead>
<tr>
<th><em><strong>Attributes</strong></em></th>
<th></th>
<th></th>
<th></th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>File Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>geftool_ver</td>
<td>uint32[3]</td>
<td>0,7,11</td>
<td>geftool version. It can be used as an individual tool to manipulate GEF files.</td>
</tr>
<tr>
<td>offsetX</td>
<td>int32</td>
<td>0</td>
<td>Minimum x coordinate in bin 1.</td>
</tr>
<tr>
<td>offsetY</td>
<td>int32</td>
<td>0</td>
<td>Minimum y coordinate in bin 1.</td>
</tr>
<tr>
<td>omics</td>
<td>S32</td>
<td>b&#x2018;Transcriptomcis&#x2019;</td>
<td>Omics name.</td>
</tr>
<tr>
<td>resolution</td>
<td>uint32</td>
<td>500</td>
<td>Pitch (nm) between neighbor spots.</td>
</tr>
<tr>
<td>version</td>
<td>uint32</td>
<td>2</td>
<td>Gene expression file format version.</td>
</tr>
<tr>
<td>bin_type</td>
<td>S32</td>
<td>CellBin</td>
<td>Bin type of the GEF file.</td>
</tr>
<tr>
<td>sn</td>
<td>S32</td>
<td>b&apos;SS200000135TL_D1&apos;</td>
<td>Stereo-seq chip SN</td>
</tr>
<tr>
<td><em><strong>/cellBin/cell:Dataset &quot;cell&quot; is a 1D array which stores basic information and indices information of cells and expression.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>averageArea</td>
<td>float32</td>
<td>494.666</td>
<td>Average area for cells in pixel.</td>
</tr>
<tr>
<td>averageDnbCount</td>
<td>float32</td>
<td>194.299</td>
<td>Average number of mRNA-captured DNBs in a cell.</td>
</tr>
<tr>
<td>averageExpCount</td>
<td>float32</td>
<td>541.715</td>
<td>Average MID count in cell.</td>
</tr>
<tr>
<td>averageGeneCount</td>
<td>float32</td>
<td>310.157</td>
<td>Average gene count in cell.</td>
</tr>
<tr>
<td>maxArea</td>
<td>uint16</td>
<td>1925</td>
<td>Maximum area for cells in pixel.</td>
</tr>
<tr>
<td>maxDnbCount</td>
<td>uint16</td>
<td>883</td>
<td>Maximum number of mRNA-captured DNBs in a cell.</td>
</tr>
<tr>
<td>maxExpCount</td>
<td>uint16</td>
<td>3018</td>
<td>Maximum MID count in cell.</td>
</tr>
<tr>
<td>maxGeneCount</td>
<td>uint16</td>
<td>1415</td>
<td>Maximum gene count in cell.</td>
</tr>
<tr>
<td>maxX</td>
<td>int32</td>
<td>17658</td>
<td>Maximum x coordinate of the cell&#x2019;s center of mass.</td>
</tr>
<tr>
<td>maxY</td>
<td>int32</td>
<td>19422</td>
<td>Maximum y coordinate of the cell&#x2019;s center of mass.</td>
</tr>
<tr>
<td>medianArea</td>
<td>float32</td>
<td>474.</td>
<td>Median area for cells in pixel.</td>
</tr>
<tr>
<td>medianDnbCount</td>
<td>float32</td>
<td>183.</td>
<td>Median number of mRNA-captured DNBs in a cell.</td>
</tr>
<tr>
<td>medianExpCount</td>
<td>float32</td>
<td>491.</td>
<td>Median MID count in cell.</td>
</tr>
<tr>
<td>medianGeneCount</td>
<td>float32</td>
<td>289.</td>
<td>Median gene count in cell.</td>
</tr>
<tr>
<td>minArea</td>
<td>uint16</td>
<td>2</td>
<td>Minimum area for cells in pixel.</td>
</tr>
<tr>
<td>minDnbCount</td>
<td>uint16</td>
<td>0</td>
<td>Minimum number of mRNA-captured DNBs in a cell.</td>
</tr>
<tr>
<td>minExpCount</td>
<td>uint16</td>
<td>0</td>
<td>Minimum MID count in cell.</td>
</tr>
<tr>
<td>minGeneCount</td>
<td>uint16</td>
<td>0</td>
<td>Minimum gene count in cell.</td>
</tr>
<tr>
<td>minX</td>
<td>int32</td>
<td>2933</td>
<td>Minimum x coordinate of the cell&#x2019;s center of mass.</td>
</tr>
<tr>
<td>minY</td>
<td>int32</td>
<td>5568</td>
<td>Minimum y coordinate of the cell&#x2019;s center of mass.</td>
</tr>
<tr>
<td><strong>Dataset DataType:compound</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>id</td>
<td>uint32</td>
<td>10</td>
<td>Cell ID index, the start ID is 0.In the Example, 10 represents the 10th cell in the dataset.</td>
</tr>
<tr>
<td>x</td>
<td>int32</td>
<td>541</td>
<td>The x coordinate of the cell&#x2019;s center of mass.In the Example, the x coordinate of the 10th cell&#x2019;s center of mass is 541.</td>
</tr>
<tr>
<td><p>y<br></p></td>
<td>int32</td>
<td>190</td>
<td>The y coordinate of the cell&#x2019;s center of mass.In the Example, the x coordinate of the 10th cell&#x2019;s center of mass is 190.</td>
</tr>
<tr>
<td>offset</td>
<td>uint32</td>
<td>494</td>
<td>The start row index of the cell in the &quot;/cellBin/cellExp&quot; dataset.The example represents that the gene ID index and total MID count information of the 10th cell in the &quot;/cellBin/cellExp&quot; dataset start from the 494th row.</td>
</tr>
<tr>
<td>geneCount</td>
<td>uint16</td>
<td>100</td>
<td>Gene count in the cell.In the example, 100 represents that the 100 rows in the &quot;/cellBin/cellExp&quot;, start from the 494th to the 593th row, contains the gene ID indices and total MID count of the gene for the 10th cell in &quot;/cellBin/cell&quot; dataset.</td>
</tr>
<tr>
<td>expCount</td>
<td>uint16</td>
<td>500</td>
<td>Cell MID count.</td>
</tr>
<tr>
<td>dnbCount</td>
<td>uint16</td>
<td>200</td>
<td>mRNA-captured DNBs of the cell.</td>
</tr>
<tr>
<td>area</td>
<td>uint16</td>
<td>474</td>
<td>Cell area in pixel.</td>
</tr>
<tr>
<td>cellTypeID</td>
<td>uint32</td>
<td>0</td>
<td>Cell type ID.</td>
</tr>
<tr>
<td>clusterID</td>
<td>uint32</td>
<td>20</td>
<td>Cell cluster ID.</td>
</tr>
<tr>
<td><em><strong>/cellBin/cellBorder:Dataset &quot;cellBorder&quot; is a 3D array which stores the lists of points for the bounding polygons of the cell.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>maxX</td>
<td>int32</td>
<td>16127</td>
<td>Maximum x coordinate of the bounding box of the cell.</td>
</tr>
<tr>
<td>maxY</td>
<td>int32</td>
<td>16663</td>
<td>Maximum y coordinate of the bounding box of the cell.</td>
</tr>
<tr>
<td>minX</td>
<td>int32</td>
<td>11129</td>
<td>Minimum x coordinate of the bounding box of the cell.</td>
</tr>
<tr>
<td>minY</td>
<td>int32</td>
<td>12784</td>
<td>Minimum y coordinate of the bounding box of the cell.</td>
</tr>
<tr>
<td><strong>Dataset DataType:3D array</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td><p><br></p></td>
<td>32*(int16,int16)</td>
<td>[[-17,-11],[-15,-5]&#x2026;[32767,32767]]</td>
<td>A list of 32 coordinates recording the differences between cell bounding points and the cell&#x2019;s center of mass (0,0). The real coordinate of cell&#x2019;s center of mass (x, y) can be obtained from &quot;cell&quot; dataset using cellID.</td>
</tr>
<tr>
<td><em><strong>/cellBin/cellExp:Dataset &quot;cellExp&quot; is a 1D array which stores the expression information of each cell.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>maxCount</td>
<td>uint16</td>
<td>336</td>
<td>Maximum MID count of a gene in a cell.</td>
</tr>
<tr>
<td><strong>Dataset DataType:compound</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>geneID</td>
<td>uint32</td>
<td>1610</td>
<td>Gene IDs of the genes detected in the cell. ID is the index of &quot;gene&quot; dataset.In the example, 1610 represents the 1610th item in the &quot;gene&quot; dataset, and the name of the gene can be acquired in &quot;gene&quot; dataset.</td>
</tr>
<tr>
<td>count</td>
<td><p>uint16<br></p></td>
<td>3</td>
<td>MID count for the gene.In the example, (assume this is the 0th item in the &quot;cellExp&quot; dataset, from the &quot;offset&quot; and &quot;geneCount&quot; record in the &quot;cell&quot; dataset we can know that the 0th item in the &quot;cellExp&quot; belongs to the cell whose cellID=0) the MID count for the gene (geneID=1610) in the cell (cellID=0) is 3.</td>
</tr>
<tr>
<td><em><strong>[optional] /cellBin/cellExon:Dataset &quot;cellExon&quot; is a 1D array which stores the exon information for each cell.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>maxExon</td>
<td>uint16</td>
<td>5793</td>
<td>Maximum exon count of a gene in all cells.</td>
</tr>
<tr>
<td>minExon</td>
<td>uint16</td>
<td>0</td>
<td>Minimum exon count of a gene in all cells.</td>
</tr>
<tr>
<td><strong>Dataset DataType:1D array</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td><p><br></p></td>
<td>uint16</td>
<td>16</td>
<td>Exon count in a cell, the index of the array is same to the cellID in the &quot;cell&quot; dataset.</td>
</tr>
<tr>
<td><em><strong>[optional] /cellBin/cellExpExon:Dataset &quot;cellExpExon&quot; is a 1D array which stores exon expression information for each cell.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>maxExon</td>
<td>uint16</td>
<td>336</td>
<td>Maximum exon count of a gene in a cell.</td>
</tr>
<tr>
<td><strong>Dataset DataType:1D array</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td><p><br></p></td>
<td>uint16</td>
<td>3</td>
<td>Exon count (MID) for the gene. The index is same to the &quot;cellExp&quot; dataset.In the example, (assume this is the 0th item in the &quot;cellExpExon&quot; dataset, since the index is same to &quot;cellExp&quot; dataset, from the &quot;offset&quot; and &quot;geneCount&quot; record in the &quot;cell&quot; dataset we can know that the 0th item in the &quot;cellExpExon&quot; belongs to the cell whose cellID=0) the exon count (MID) for the gene (geneID=1610) in the cell (cellID=0) is 3.</td>
</tr>
<tr>
<td><em><strong>/cellBin/cellTypeList:Dataset &quot;cellTypeList&quot; is a 1D array which stores cell types of each cell.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset DataType:1D array</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td><p><br></p></td>
<td>S32</td>
<td>b&apos;default&apos;</td>
<td>Cell type, &quot;default&quot; stands for undefined cell type.</td>
</tr>
<tr>
<td><strong>/cellBin/gene:Dataset &quot;gene&quot; is a 1D array which stores the indices of cell and expression information of each gene.</strong></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>maxCellCount</td>
<td>uint32</td>
<td>5718</td>
<td>Maximum number of cells a gene can be detected.</td>
</tr>
<tr>
<td>maxExpCount</td>
<td>uint32</td>
<td>55361</td>
<td>Maximum MID count of a gene.</td>
</tr>
<tr>
<td>minCellCount</td>
<td>uint32</td>
<td>1</td>
<td>Minimum number of cells a gene can be detected.</td>
</tr>
<tr>
<td>minExpCount</td>
<td>uint32</td>
<td>1</td>
<td>Minimum MID count of a gene.</td>
</tr>
<tr>
<td><strong>Dataset DataType:compound</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>geneID</td>
<td>S32</td>
<td>b&apos;ENSMUSG00000000001&apos;</td>
<td>Gene ID.</td>
</tr>
<tr>
<td>geneName</td>
<td>S32</td>
<td>b&apos;AC149090.1&apos;</td>
<td>Gene name.</td>
</tr>
<tr>
<td>offset</td>
<td>uint32</td>
<td>0</td>
<td>The start row index of the gene in &quot;/cellBin/geneExp&quot; dataset.In the example, 0 means that start from the 0th item in &quot;/cellBin/geneExp&quot; dataset records the cellIDs and total MID count information of &quot;AC149090.1&quot;.</td>
</tr>
<tr>
<td>cellcount</td>
<td>uint32</td>
<td>60</td>
<td>Number of cells a gene can be detected.In the example, 60 represents that start from the 0th item to the 59th item records the information of gene &quot;AC149090.1&quot;.</td>
</tr>
<tr>
<td>expCount</td>
<td>uint32</td>
<td>100</td>
<td>Sum of MID count for the gene.In the example, the total MID count of &quot;AC149090.1&quot; is 100.</td>
</tr>
<tr>
<td>maxMIDcount</td>
<td>uint16</td>
<td>4</td>
<td>Maximum MID count of a gene in a cell.In this case, the maximum MID count of gene &quot;AC149090.1&quot; in a cell is 4.</td>
</tr>
<tr>
<td><em><strong>/cellBin/geneExp:Dataset &quot;geneExp&quot; is a 1D array which stores cell and expression information of each gene.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>maxCount</td>
<td>uint16</td>
<td>10</td>
<td>Maximum MID count of a gene.</td>
</tr>
<tr>
<td><strong>Dataset DataType:compound</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>cellID</td>
<td>uint32</td>
<td>1247</td>
<td>cellID that contains the gene whose index is same to the index in &quot;gene&quot; dataset.In the example, (assume we use the 0th item in &quot;geneExp&quot; dataset) 1247 shows that the gene &quot;AC149090.1&quot; appears in the cell whose cellID is 1247.</td>
</tr>
<tr>
<td>count</td>
<td>uint16</td>
<td>3</td>
<td>The MID count of the gene, whose index is same to the index in &quot;gene&quot; dataset, in the cellID.In the example, the MID count of gene &quot;AC149090.1&quot; in the cell (cellID=1247) is 3.</td>
</tr>
<tr>
<td><em><strong>[optional] /cellBin/geneExon:Dataset &quot;geneExon&quot; is a 1D array which stores the exon expression information of each gene.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>maxExon</td>
<td>uint32</td>
<td>55361</td>
<td>Maximum exon count of a gene.</td>
</tr>
<tr>
<td>minExon</td>
<td>uint32</td>
<td>0</td>
<td>Minimum exon count of a gene.</td>
</tr>
<tr>
<td><strong>Dataset DataType:1D array</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td><p><br></p></td>
<td>uint32</td>
<td>97</td>
<td>Total exon count of a gene, the index of &quot;geneExon&quot; dataset is same to the &quot;gene&quot; dataset.In the example, (assume this is the 0th item in the &quot;geneExon&quot; dataset, and gene &quot;AC149090.1&quot; is the 0th item in the &quot;gene&quot; dataset) the exon count of gene &quot;AC149090.1&quot; is 97.</td>
</tr>
<tr>
<td><em><strong>[optional] /cellBin/geneExpExon:Dataset &quot;geneExpExon&quot; is a 1D array which stores the exon expression information in cells of each gene.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td>maxExon</td>
<td>uint16</td>
<td>336</td>
<td>Maximum exon expression of a gene in a cell.</td>
</tr>
<tr>
<td><strong>Dataset DataType:1D array</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td><p><br></p></td>
<td>uint16</td>
<td>3</td>
<td>Exon count of a gene in a cell. The index of &quot;geneExpExon&quot; dataset is same to the &quot;geneExp&quot; dataset.In the example, (assume this is the 0th item in the &quot;geneExpExon&quot; dataset, since the index is same to &quot;geneExp&quot; dataset, from the &quot;offset&quot; and &quot;cellCount&quot; record in the &quot;gene&quot; dataset we can know that the 0th item in the &quot;geneExpExon&quot; dataset belongs to the gene &quot;AC149090.1&quot;) 3 stands for the exon count of gene &quot;AC149090.1&quot; in cell 1247 is 3.</td>
</tr>
<tr>
<td><em><strong>/cellBin/bockIndex:Dataset &quot;bockIndex&quot; is a 1D array which stores the matrix block partition information.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset DataType:1D array</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td><p><br></p></td>
<td>uint32</td>
<td>0</td>
<td>Cell count in each partition block.cnt=blockIndex[i+1]-blockIndex[i]</td>
</tr>
<tr>
<td><em><strong>/cellBin/bockSize:Dataset &quot;bockSize&quot; is a 1D array which stores the block size of partition.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset DataType:1D array</strong></td>
<td><p><strong>DataType</strong><br></p></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td><p><br></p></td>
<td>uint32</td>
<td>256, 256, 104, 104</td>
<td>4-element array. The 4 items represent the block length in x-axis, block length in y-axis, block count in x-axis, and block count in y-axis, respectively.</td>
</tr>
<tr>
<td><em><strong>[optional] /codedCellBlock:Group &quot;codedCellBlock&quot; stores pre-computed data for rendering in StereoMap.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Group Attributes</strong></td>
<td><strong>DataType</strong></td>
<td><strong>Example</strong></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td><p>info<br></p></td>
<td>string</td>
<td>{&quot;@type&quot;: &quot;neuroglancer_annotations_v1&quot;, ...}</td>
<td><p>Metadata of encoded precomputed data in JSON.<br></p></td>
</tr>
<tr>
<td><em><strong>[optional] /codedCellBlock/L0/0_1:Dataset &quot;0_1&quot; is an example chunk encoded pre-computed data, including id, geometry, and so on.</strong></em></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr>
<td><strong>Dataset DataType:Bytes</strong></td>
<td><p><strong>DataType</strong><br></p></td>
<td><p><strong>Example</strong><br></p></td>
<td><strong>Description</strong></td>
</tr>
<tr>
<td><p><br></p></td>
<td>H5T_OPAQUE</td>
<td>1F 8B 08 00 ...</td>
<td>Bytecode of the chunk.</td>
</tr>
</tbody>
</table>
<h2 id="gene-expression-matrix-gem"><a name="gene-expression-matrix-gem" class="anchor-navigation-ex-anchor" href="#gene-expression-matrix-gem"><i class="fa fa-link" aria-hidden="true"></i></a>Gene Expression Matrix (GEM)</h2>
<p>Gene expression matrix (GEM) stores gene spatial expression data. SAW generates multiple gene expression matrix files in the workflow, the basic format requires six columns with a header row that shows the column names. The six columns are gene ID, gene name, x coordinate, y coordinate, MID count and exon count. When it comes to cellbin GEM, the seventh column is for cell ID. The header of the expression matrix for the maximum area enclosing rectangle region has several annotation rows starting with &quot;#&quot; before the column rows. The header field names and field types are described in the table.</p>
<table>
<thead>
<tr>
<th>Fields</th>
<th>Data Type</th>
<th>Example</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>#FileFormat</td>
<td>string</td>
<td>GEMv0.2</td>
<td>Gene expression matrix file format version.</td>
</tr>
<tr>
<td>#SortedBy</td>
<td>string</td>
<td><p>None<br></p></td>
<td>Gene expression matrix sorting strategy. Valid values: &quot;geneID&quot;, &quot;x&quot;, &quot;y&quot;, &quot;MIDCount&quot;, &quot;None&quot;.</td>
</tr>
<tr>
<td>#BinType</td>
<td>string</td>
<td>Bin</td>
<td>Bin type of the GEM file.</td>
</tr>
<tr>
<td>#BinSize</td>
<td>string</td>
<td>1</td>
<td>(Please check 1.3 Terminologies and Concepts Bin)</td>
</tr>
<tr>
<td>#Omics</td>
<td>string</td>
<td>Transcriptomics</td>
<td>Omics name.</td>
</tr>
<tr>
<td>#Stereo-seqChip</td>
<td>string</td>
<td>SS200000135TL_D1</td>
<td>Stereo-seq Chip T serial number.</td>
</tr>
<tr>
<td>#OffsetX</td>
<td>uint32</td>
<td>1</td>
<td>X coordinate of the origin before calibration.</td>
</tr>
<tr>
<td>#OffsetY</td>
<td>uint32</td>
<td>1</td>
<td>Y coordinate of the origin before calibration.</td>
</tr>
<tr>
<td>geneID</td>
<td>string</td>
<td>ENSMUSG00000000001</td>
<td>Gene ID</td>
</tr>
<tr>
<td>geneName</td>
<td>string</td>
<td>Gnai3</td>
<td>Gene name.</td>
</tr>
<tr>
<td>x</td>
<td>uint32</td>
<td>16809</td>
<td>X coordinate of the spot.</td>
</tr>
<tr>
<td>y</td>
<td>uint32</td>
<td>8546</td>
<td>Y coordinate of the spot.</td>
</tr>
<tr>
<td>MIDCount</td>
<td>uint32</td>
<td>1</td>
<td>Number of MIDs at (x, y) for the gene in the corresponding row.</td>
</tr>
<tr>
<td>ExonCount</td>
<td>uint32</td>
<td>0</td>
<td>[Optional] Number of exon count at (x, y) for the gene in the corresponding row.</td>
</tr>
<tr>
<td>CellID</td>
<td>uint32</td>
<td>55892</td>
<td>[Optional] CellID for (x, y).</td>
</tr>
</tbody>
</table>
<p>An example of bin GEM:</p>
<pre><code class="lang-Bash"><span class="hljs-comment">#FileFormat=GEMv0.2</span>
<span class="hljs-comment">#SortedBy=None</span>
<span class="hljs-comment">#BinType=Bin</span>
<span class="hljs-comment">#BinSize=1</span>
<span class="hljs-comment">#Omics=Transcriptomics</span>
<span class="hljs-comment">#Stereo-seqChip=B03523G1</span>
<span class="hljs-comment">#OffsetX=0</span>
<span class="hljs-comment">#OffsetY=0</span>
geneID  geneName        x       y       MIDCount        ExonCount
ENSMUSG00000000001      Gnai3   694     17229   1       1
ENSMUSG00000000001      Gnai3   1428    4994    1       1
</code></pre>
<p>An example of cellbin GEM:</p>
<pre><code>#FileFormat=GEMv0.2
#SortedBy=None
#BinType=CellBin
#BinSize=Cell
#Omics=Transcriptomics
#Stereo-seqChip=B03523G1
#OffsetX=0
#OffsetY=0
geneID  geneName        x       y       MIDCount        ExonCount       CellID
ENSMUSG00000047454      Gphn    9325    19972   1       0       192276
ENSMUSG00000030616      Sytl2   9314    19976   1       1       192276
</code></pre><footer class="page-footer"><span class="copyright">&#xA9; 2024 STOmics Tech. All right reserved.</span><span class="footer-modification">Modiftied: 
2024-07-22 14:57:47
</span></footer><script>document.addEventListener("DOMContentLoaded", function() {lightbox.option({"resizeDuration":200,"wrapAround":true});})</script>
                                
                                </section>
                            
    </div>
    <div class="search-results">
        <div class="has-results">
            
            <h1 class="search-results-title"><span class='search-results-count'></span> results matching "<span class='search-query'></span>"</h1>
            <ul class="search-results-list"></ul>
            
        </div>
        <div class="no-results">
            
            <h1 class="search-results-title">No results matching "<span class='search-query'></span>"</h1>
            
        </div>
    </div>
</div>

                        </div>
                    </div>
                
            </div>

            
                
                <a href="analysis-with-gpu.html" class="navigation navigation-prev " aria-label="Previous page: Analysis with GPU">
                    <i class="fa fa-angle-left"></i>
                </a>
                
                
                <a href="image-related-record-format.html" class="navigation navigation-next " aria-label="Next page: Image-related record format">
                    <i class="fa fa-angle-right"></i>
                </a>
                
            
        
    </div>

    <script>
        var gitbook = gitbook || [];
        gitbook.push(function() {
            gitbook.page.hasChanged({"page":{"title":"Expression matrix format","level":"1.7.2","depth":2,"next":{"title":"Image-related record format","level":"1.7.3","depth":2,"path":"gao-ji-she-zhi/image-related-record-format.md","ref":"gao-ji-she-zhi/image-related-record-format.md","articles":[]},"previous":{"title":"Analysis with GPU","level":"1.7.1","depth":2,"path":"gao-ji-she-zhi/analysis-with-gpu.md","ref":"gao-ji-she-zhi/analysis-with-gpu.md","articles":[]},"dir":"ltr"},"config":{"plugins":["expandable-chapters-small","anchor-navigation-ex","-insert-logo","tbfed-pagefooter","hide-element","code-optimize","katex","hints","tabs","lightbox","-lunr","-search","search-plus","add-index-html","livereload"],"root":".","styles":{},"pluginsConfig":{"tbfed-pagefooter":{"copyright":"© 2024 STOmics Tech.","modify_label":"Modiftied: ","modify_format":"YYYY-MM-DD HH:mm:ss"},"livereload":{},"hints":{"danger":"fa fa-exclamation-circle","info":"fa fa-info-circle","tip":"fa fa-mortar-board","working":"fa fa-wrench"},"hide-element":{"elements":[".gitbook-link"]},"katex":{},"add-index-html":{"myConfigKey":"it's the default value","elements":["a"]},"fontsettings":{"theme":"white","family":"sans","size":2},"highlight":{},"anchor-navigation-ex":{"associatedWithSummary":true,"float":{"floatIcon":"fa fa-navicon","level1Icon":"fa fa-hand-o-right","level2Icon":"fa fa-hand-o-right","level3Icon":"fa fa-hand-o-right","showLevelIcon":false},"mode":"float","multipleH1":true,"pageTop":{"level1Icon":"fa fa-hand-o-right","level2Icon":"fa fa-hand-o-right","level3Icon":"fa fa-hand-o-right","showLevelIcon":false},"printLog":true,"showGoTop":true,"showLevel":false},"lightbox":{"jquery":true,"includeJQuery":true,"sameUuid":true,"options":{"resizeDuration":200,"wrapAround":true}},"expandable-chapters-small":{},"sharing":{"facebook":false,"twitter":false,"google":false,"weibo":false,"instapaper":false,"vk":false,"all":[]},"theme-default":{"styles":{"website":"styles/website.css","pdf":"styles/pdf.css","epub":"styles/epub.css","mobi":"styles/mobi.css","ebook":"styles/ebook.css","print":"styles/print.css"},"showLevel":false},"code-optimize":{"copyButtons":true,"showLines":true},"tabs":{},"search-plus":{}},"theme":"default","author":"","pdf":{"pageBreaksBefore":"/","headerTemplate":null,"paperSize":"a4","margin":{"right":62,"left":62,"top":36,"bottom":36},"fontSize":12,"fontFamily":"Arial","footerTemplate":null,"chapterMark":"pagebreak","pageNumbers":false},"structure":{"langs":"LANGS.md","readme":"README.md","glossary":"GLOSSARY.md","summary":"SUMMARY.md"},"isbn":"","variables":{},"title":"SAW User Manual","language":"zh-hans","links":{"gitbook":false,"sharing":{"google":false,"facebook":false,"twitter":false,"all":false}},"gitbook":"3.2.3","description":null,"extension":null},"file":{"path":"gao-ji-she-zhi/expression-matrix-format.md","mtime":"2024-07-22T06:57:47.388Z","type":"markdown"},"gitbook":{"version":"3.2.3","time":"2024-07-22T07:44:47.860Z"},"basePath":"..","book":{"language":""}});
        });
    </script>
</div>

        
    <script src="../gitbook/gitbook.js"></script>
    <script src="../gitbook/theme.js"></script>
    
        
        <script src="../gitbook/gitbook-plugin-expandable-chapters-small/expandable-chapters-small.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-hide-element/plugin.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-code-optimize/plugin.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-tabs/bootstrap.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-lightbox/js/lightbox.min.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-search-plus/jquery.mark.min.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-search-plus/search.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-add-index-html/plugin.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-livereload/plugin.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-sharing/buttons.js"></script>
        
    
        
        <script src="../gitbook/gitbook-plugin-fontsettings/fontsettings.js"></script>
        
    

    </body>
</html>

